<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
    </head>
    <body>
        




<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Transcriptomics
                        </a>
                    </li>

                    
                        
                        
                        
                            <li class="nav-item dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Introduction slides">
                                    <i class="fa fa-slideshare" aria-hidden="true"></i> Introduction slides
                                </a>
                                <div class="dropdown-menu">
                                    
                                        
                                            
                                                <a class="dropdown-item" href="/training-material/topics/transcriptomics/slides/introduction.html">
                                                    Introduction to Transcriptomics
                                                </a>
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                </div>
                            </li>
                        
                    

                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="https://zenodo.org/record/1001880" title="Links to data">
                            <i class="fa fa-files-o" aria-hidden="true"></i> Input Dataset
                        </a>
                    </li>
                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i> Help
    </a>
    <div class="dropdown-menu">
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Gitter
        </a>
        <a class="dropdown-item" href="https://biostar.usegalaxy.org/" title="Ask on Biostars">
            Biostars
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>Visualization of RNA-Seq results with CummeRbund</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>How are RNA-Seq results stored?</li>
            
            <li>Why are visualization techniques needed?</li>
            
            <li>How to select our desired subjects for differential gene expression analysis?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>Manage RNA-Seq results</li>
            
            <li>Extract the desired subject for differential gene expression analysis</li>
            
            <li>Visualize information</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="/training-material/topics//introduction/">Galaxy introduction</a>
                    
                </li>
            
                <li>
                    
                    <a href="/training-material/topics//sequence-analysis/">Quality control</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 1h</p>
        </blockquote>

        <h1 class="no_toc" id="introduction">Introduction</h1>

<p>RNA-Seq analysis helps researchers annotate new genes and splice variants, and provides cell- and context-specific quantification of gene expression. RNA-Seq data, however, are complex and require both computer science and mathematical knowledge to be managed and interpreted.</p>

<p>Visualization techniques are key to overcome the complexity of RNA-Seq data, and represent valuable tools to gather information and insights.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#reasons-for-visualizing-rna-seq-results" id="markdown-toc-reasons-for-visualizing-rna-seq-results">Reasons for visualizing RNA-Seq results</a></li>
  <li><a href="#importing-rna-seq-result-data" id="markdown-toc-importing-rna-seq-result-data">Importing RNA-Seq result data</a></li>
  <li><a href="#filtering-and-sorting" id="markdown-toc-filtering-and-sorting">Filtering and sorting</a></li>
  <li><a href="#cummerbund" id="markdown-toc-cummerbund">CummeRbund</a></li>
</ol>

</blockquote>

<h1 id="reasons-for-visualizing-rna-seq-results">Reasons for visualizing RNA-Seq results</h1>

<p>To make sense of the available RNA-Seq data, and overview the condition-specific gene expression levels of the provided samples, we need to visualize our results. Here we will use <a href="http://compbio.mit.edu/cummeRbund/">CummeRbund</a>.</p>

<p>CummeRbund is an open-source tool that simplifies the analysis of a CuffDiff RNA-Seq output. In particular, it helps researchers:</p>
<ul>
  <li>managing, integrating, and visualizing the data produced by CuffDiff</li>
  <li>simplifying data exploration</li>
  <li>providing a bird’s-eye view of the expression analysis by describing relationships betweeen genes, transcripts, transcription start sites, and protein-coding regions</li>
  <li>exploring subfeatures of individual genes or gene-sets</li>
  <li>creating publication-ready plots</li>
</ul>

<p>A typical workflow for the visualization of RNA-Seq data involving CummeRbund:</p>

<p><img src="../../images/cummerbund-rna-seq-viz-with-cummerbund.png" alt="workflow" /></p>

<p>CummeRbund reads your RNA-Seq results from a <a href="https://www.sqlite.org/">SQLite</a> database. This database has to be created using CuffDiff’s SQLite output option.</p>

<blockquote class="tip">
  <h3 id="-tip-sqlite-output-with-cuffdiff"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: SQLite output with CuffDiff</h3>

  <p>Instruct CuffDiff to organize its output in a SQLite database to be read CummeRbund.</p>

  <p><img src="../../images/cummerbund-cuffdiff-set-sqlite.png" alt="SQLite output" /></p>
</blockquote>

<h1 id="importing-rna-seq-result-data">Importing RNA-Seq result data</h1>

<blockquote class="hands_on">
  <h3 id="-hands-on-data-upload"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history</li>
    <li>
      <p>Import the <a href="https://zenodo.org/record/1001880/files/CuffDiff_SQLite_database.sqlite"><code class="highlighter-rouge">CuffDiff SQLite</code></a> dataset</p>

      <blockquote class="tip">
        <h3 id="-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>Paste the link into the text field</li>
          <li>Press <strong>Start</strong></li>
        </ul>
      </blockquote>

      <blockquote class="comment">
        <h3 id="-comments"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>

        <p>Rename the dataset to “RNA-Seq SQLite result data”</p>
      </blockquote>
      <p>By default, when data is imported via its link, Galaxy names it with its URL.</p>
    </li>
  </ol>
</blockquote>

<p>CuffDiff’s output data is organized in a SQLite database, so we need to extract it to be able to see what it looks like.</p>

<p>For this tutorial, we are interested in CuffDiff’s tested transcripts for differential expression.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-extract-cuffdiff-results"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Extract CuffDiff results</h3>

  <ol>
    <li><strong>Extract CuffDiff</strong> <i class="fa fa-wrench" aria-hidden="true"></i> with the following parameters
      <ul>
        <li>“Select tables to output” to <code class="highlighter-rouge">Transcript differential expression testing</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the table</p>

      <blockquote class="tip">
        <h3 id="-tip-inspecting-the-content-of-a-file-in-galaxy"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Inspecting the content of a file in Galaxy</h3>

        <ul>
          <li>Click on the eye (“View data”) on the right of the file name in the history</li>
          <li>Inspect the content of the file on the middle</li>
        </ul>
      </blockquote>
    </li>
  </ol>

  <p>Each entry represents a differentially expressed gene, but not all are significant. We want to keep only those that are reported as significant differentially expressed.</p>

  <blockquote class="question">
    <h3 id="-questions"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

    <ol>
      <li>How to retain only the significant differentially expressed genes?</li>
      <li>Which column stores this information?</li>
    </ol>

    <details>
<summary>Click to view the answers</summary>
<ol type="1">
<li>We need to filter on the column storing the record's significance</li>
<li>Column 14</li>
</ol>
</details>
  </blockquote>
</blockquote>

<h1 id="filtering-and-sorting">Filtering and sorting</h1>

<p>We now want to first highlight the most significant differentially expressed genes in our analysis, and then obtain informative visualizations.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-extract-cuffdiffs-most-significant-differentially-expressed-genes"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Extract CuffDiff’s most significant differentially expressed genes</h3>

  <ol>
    <li><strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i> with the following parameters
      <ul>
        <li>“Filter” to the extracted table from the previous step</li>
        <li>“With following condition” to an appropriate filter over the target column (see questions below when in doubt)</li>
      </ul>

      <blockquote class="question">
        <h3 id="-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>
        <ol>
          <li>What column stores the information of significance for each record?</li>
          <li>Which conditional expression has to be set to filter all records on the selected column?</li>
          <li>What happened to the records in the original table?</li>
        </ol>

        <details>
<summary>Click to view the answers</summary>
<ol type="1">
<li>column 14</li>
<li>c14=='yes'</li>
<li>All records whose "significant" field was set to "yes" have been retained, while the others filtered out</li>
</ol>
</details>
      </blockquote>
    </li>
  </ol>

  <p>Look at your data. The differential expression values are stored on column 10, we will sort (descending) all records
 on the basis of their value at the 10th column</p>

  <ol>
    <li><strong>Sort</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: with the following parameters
      <ul>
        <li>“Sort Dataset” to the filtered table</li>
        <li>“on column”, “with flavor” and “everything in”  to the appropriate values (see above)</li>
      </ul>

      <blockquote class="question">
        <h3 id="-questions-2"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>
        <ol>
          <li>Since the start of our filtering process, how many records now represent the significant subset for extracting informations?</li>
          <li>What does this shrinking of the number of lines represent?</li>
        </ol>

        <details>
<summary>Click to view the answers</summary>
<ol type="1">
<li>Click on the boxes in your history, their small preview higlights the number of lines: from ~140,000 to 219</li>
<li>This process represents a necessary step to gather insights on the biological meaning of our samples in our analyses: putting the original raw RNA-Seq result data into context, cutting down the less-meaningful records to focus on what is needed to go from data to information</li>
</ol>
</details>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="cummerbund">CummeRbund</h1>

<p>With CummeRbund we can visualize our RNA-Seq results of interest.</p>

<p>CummeRbund generates always two outputs:</p>
<ul>
  <li>the plot</li>
  <li>the R script responsible for generating the plot</li>
</ul>

<p>We are interested in visualizing all expression values of all transcripts relative to the most significant differentially expressed gene we found in the previous section.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-visualization"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Visualization</h3>

  <ol>
    <li><strong>CummeRbund</strong> <i class="fa fa-wrench" aria-hidden="true"></i> with the following parameters
      <ul>
        <li>Click on “Insert plot”</li>
        <li>“Width” and “Height” to <code class="highlighter-rouge">800x600</code></li>
        <li>“Plot type” to <code class="highlighter-rouge">Expression Plot</code></li>
        <li>“Expression levels to plot” to <code class="highlighter-rouge">Isoforms</code></li>
        <li>“Gene ID” to <code class="highlighter-rouge">NDUFV1</code></li>
        <li>Your input form parameters should look like the following. If so, click on “Execute”</li>
      </ul>
    </li>
  </ol>

  <p><img src="../../images/cummerbund-expression-plot-form.png" alt="Expression plot_form" /></p>
</blockquote>

<p>Our first CummeRbund plot is the “Expression Plot”:</p>

<p><img src="../../images/cummerbund-expression-plot.png" alt="Expression plot" /></p>

<p>The Expression Plot represents the expression of all isoforms of a single gene (NDUFV1) with replicate FPKMs exposed.</p>

<p>Our plot has a modest number of isoforms, and is therefore already readable. However, in case of 5 or 6 isoforms, the plot can look very busy. We can therefore change the visualization type by selecting another type of plot.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-visualization-1"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Visualization</h3>

  <ol>
    <li><strong>CummeRbund</strong> <i class="fa fa-wrench" aria-hidden="true"></i> with the following parameters
      <ul>
        <li>Click on “Insert plot”</li>
        <li>“Width” and “Height” to <code class="highlighter-rouge">800x600</code></li>
        <li>“Plot type” to <code class="highlighter-rouge">Expression Bar Plot</code></li>
        <li>“Expression levels to plot” to <code class="highlighter-rouge">Isoforms</code></li>
        <li>“Gene ID” to <code class="highlighter-rouge">NDUFV1</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/cummerbund-expression-bar-plot.png" alt="Expression bar plot" /></p>

<p>Expression Bar Plot of a single gene (NDUFV1) with replicate FPKMs exposed.</p>

<blockquote class="comment">
  <h3 id="-comment"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

  <p>These plots are shown also in <a href="https://vimeo.com/channels/884356/128265982">this</a> Galaxy video tutorial.</p>

  <p>Would you like to obtain more sophisticated visualization of your RNA-Seq analysis results? Select different CummeRbund plot options, and look at their parametrizations according to the filtering and sorting operations we performed</p>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>Visualization tools help researchers making sense of data, providing a bird’s-eye view of the underlying analysis results.
In this tutorial we overviewed the advantages of visualizing RNA-Seq results with CummeRbund, and gained insights on CuffDiff’s big-data output by plotting informations relative to the most significant differentially expressed genes in our RNA-Seq analysis.</p>


        
        <blockquote class="key_points">
            <h3><i class="fa fa-key" aria-hidden="true"></i> Key points</h3>

            <ul>
                
                <li>Extract informations from a SQLite CuffDiff database</li>
                
                <li>Filter and sort results to highlight differential expressed genes of interest</li>
                
                <li>Generate publication-ready visualizations for RNA-Seq analysis results</li>
                
            </ul>
        </blockquote>
        

        
        <h1>Useful literature</h1>
        <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
        

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i> Help us improve this content!</h3>
            Please take a moment to fill in the Galaxy Training Network
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>.
            Your feedback helps us improve this tutorial and will be considered
            in future revisions.
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Andrea Bagnacani)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
</html>
