<<<<<<< HEAD
---
layout: introduction_slides
topic_name: variant-analysis
tutorial_name: introduction
logo: "GTN"
---

=======
<!DOCTYPE html>
<html>






  <head>
    <meta charset="utf-8">
    <title>Introduction to Variant analysis</title>
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css" id="theme">
  </head>
  <body>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-footer"><span>

<img src="/training-material/assets/images/GTN-60px.png" alt="Galaxy Training Network" style="height: 40px;"/>

</span></div>

---

# Introduction to Variant analysis


---

  ## Requirements

  Before diving into this slide deck, we recommend you to have a look at:

  
    
  - [Galaxy introduction](/training-material/topics//introduction/)
    
  
    
  - [Quality control](/training-material/topics//sequence-analysis/)
    
  
    
  - [Mapping](/training-material/topics//sequence-analysis/)
    
  

  
    
  


---
>>>>>>> f1b01b7ef32ec4f29a7f797f014dae1d28ea2906

# What is Exome sequencing?

---

### Exome sequencing

= Whole exome sequencing (WES or WXS)

Sequencing of all expressed protein-coding genes in a genome

---

### Exome in Humans

- ~180,000 exons
- 1% of the human genome
- ~30 million base pairs

---

### Goal of exome sequencing

Identify genetic variation that is responsible for both Mendelian
and common diseases without the high costs associated with
whole-genome sequencing

Exome sequencing is the most efficient way to identify the
genetic variants in all of an individual's genes

---

### Limits

Exome sequencing can not identify genetic variation in

- All genes
- Mitochondrial genes
- “Structural variants”
- Triplet repeat disorders
- Other copy number variants
- Introns
- “Uniparental disomy”
- Control sequences
- Epigenetic changes
- Gene-gene (epistatic) interactions

---

## 2 tutorials for training on exome sequencing data analysis

---

### Same goal

Identify and annotate genetic variants in a family with two
parents and a child exome data

---

### Similar data analysis approach

![](../images/genVAST.png)

---

 ### 2 tutorials

<<<<<<< HEAD
- [Introductory tutorial]({{site.baseurl}}/topics/variant-analysis/tutorials/exome-seq/tutorial.html)
- [Detailed tutorial]({{site.baseurl}}/topics/variant-analysis/tutorials/diploid-variant-calling/tutorial.html)
=======
- [Introductory tutorial](/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html)
- [Detailed tutorial](/training-material/topics/variant-analysis/tutorials/diploid-variant-calling/tutorial.html)


---



## Related tutorials


    

    
        
    

    
        
    

    
        
    

    
        
    

    
        
    

    
        
    

    
        
    



---

## Thank you!

This material is the result of a collaborative work. Thanks the [Galaxy Training Network](https://wiki.galaxyproject.org/Teach/GTN) and all the contributors  (Bérénice Batut, Yvan Le Bras) !


<img src="/training-material/assets/images/GTN.png" alt="Galaxy Training Network" style="height: 100px;"/>



.footnote[Found a typo? Something is wrong in this tutorial? <br/>Edit it on [GitHub](https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/slides/introduction.html)]

    </textarea>
    <script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript">
    </script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
    </script>
  </body>
</html>
>>>>>>> f1b01b7ef32ec4f29a7f797f014dae1d28ea2906
