<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
    </head>
    <body>
        




<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Transcriptomics
                        </a>
                    </li>

                    
                        
                        
                        
                            <li class="nav-item dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Introduction slides">
                                    <i class="fa fa-slideshare" aria-hidden="true"></i> Introduction slides
                                </a>
                                <div class="dropdown-menu">
                                    
                                        
                                            
                                                <a class="dropdown-item" href="/training-material/topics/transcriptomics/slides/introduction.html">
                                                    Introduction to Transcriptomics
                                                </a>
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                </div>
                            </li>
                        
                    

                    

                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i> Help
    </a>
    <div class="dropdown-menu">
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Gitter
        </a>
        <a class="dropdown-item" href="https://biostar.usegalaxy.org/" title="Ask on Biostars">
            Biostars
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/functional-enrichment/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>Functional Enrichment Analysis</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>What is enrichment analysis?</li>
            
            <li>How do I perform functional enrichment analysis?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>Learn what is enrichment analysis, gene ontology (GO) and GO annotations.</li>
            
            <li>Perform functional enrichment analysis using GOEnrichment.</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="/training-material/topics//introduction/">Galaxy introduction</a>
                    
                </li>
            
                <li>
                    
                    <a href="/training-material/topics//sequence-analysis/">Quality control</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 1h</p>
        </blockquote>

        <h1 class="no_toc" id="introduction">Introduction</h1>

<p>When we have a large list of genes of interest (for example, a list of differentially expressed genes obtained from an RNA-Seq experiment), how do we make sense of the biological meaning of that list? One way is to perform functional enrichment analysis. This method consists of the application of statistical tests to verify if genes of interest are more often associated to certain biological functions when compared with a random set of genes. In this tutorial you will learn about enrichment analysis and how to perform it.</p>

<p><strong>What is the Gene Ontology?</strong> <br />
The <a href="http://www.geneontology.org/">Gene Ontology</a> provides structured, controlled vocabularies and classifications of many domains of molecular and cellular biology. It is divided in three separate ontologies: biological process (e.g., signal transduction), molecular function (e.g., catalytic activity) and cellular component (e.g., ribosome). These ontologies are structured as a directed acyclic graph (a hierarchy with multi-parenting) connecting GO terms which represent all the different molecular and cellular functions.</p>

<p><img src="images/GOexample.png" alt="" /></p>
<blockquote>

  <p><strong>Figure 1</strong> QuickGO - http://www.ebi.ac.uk/QuickGO</p>
</blockquote>

<p><strong>What is GO annotation?</strong> <br />
Genes are associated to GO terms via annotations, where each gene can be associated to multiple annotations. An important notion to take into account when using GO is that, according to the <strong>true path rule</strong>, a gene annotated to a term is also implicitly annotated to each ancestor of that term. GO annotations have evidence codes that encode the type of evidence supporting them (eg. experimentally verified (a small minority), or inferred from in-silico experiments).</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#functional-enrichment-analysis" id="markdown-toc-functional-enrichment-analysis">Functional Enrichment Analysis</a></li>
  <li><a href="#simplification-of-graphs" id="markdown-toc-simplification-of-graphs">Simplification of graphs</a></li>
  <li><a href="#interpretation-of-the-results" id="markdown-toc-interpretation-of-the-results">Interpretation of the results</a></li>
</ol>

</blockquote>

<h1 id="functional-enrichment-analysis">Functional Enrichment Analysis</h1>

<p>To perform a functional enrichment analysis, we need to have:</p>
<ul>
  <li>A set of genes of interest (eg. differentially expressed genes): <strong>study set</strong></li>
  <li>A list with all the genes to consider in the analysis: <strong>population set</strong></li>
  <li>Gene annotations, associating genes to GO terms</li>
  <li>The GO ontology, with description of GO terms and their relationships</li>
</ul>

<p>For each GO term, we need to count how many (<strong>k</strong>) of the genes in the study set (<strong>K</strong>) are associated to the term, an how many (<strong>n</strong>) of the genes in the whole population (<strong>N</strong>) are associated to the same term. Then we test how likely would it be to obtain <strong>k</strong> genes associated to the term if <strong>K</strong> genes would be randomly sampled from the population.</p>

<p>The appropriate statistical test is the one-tailed variant of Fisher’s exact test, also known as hypergeometric test for over-representation. When the one-tailed version is applied, this test will compute the probability of observing at least the sample frequency, given the population frequency. The <a href="https://en.wikipedia.org/wiki/Hypergeometric_distribution">hypergeometric distribution</a> consists in the probability of <strong>k</strong> successes in <strong>n</strong> draws, without  replacement, from a finite population of size <strong>N</strong> that contains exactly <strong>K</strong> successful objects:</p>
<blockquote>

  <p><img src="images/formula.png" alt="" /></p>
</blockquote>

<blockquote class="hands_on">
  <h3 id="-hands-on"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <p>For this first exercise we will use data from <a href="https://www.ncbi.nlm.nih.gov/pubmed/22383036" title="Trapnell et al. data">Trapnell et al.</a>. In this work, the authors created an artificial (in-silico) dataset of gene expression in Drosophila melanogaster, where 300 random genes were set to be differentially expressed between two conditions.</p>
  <ol>
    <li>
      <p><strong>Create a new history</strong></p>
    </li>
    <li><strong>Upload to the Galaxy</strong> the following files:
      <ul>
        <li>go.obo  (LINK)</li>
        <li>drosophila_gene_association.fb (LINK)</li>
        <li>trapnellPopulation.tab (LINK)</li>
      </ul>

      <blockquote class="tip">
        <h3 id="-tip-upload-data-to-galaxy-1"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Upload data to Galaxy <a href="https://galaxyproject.github.io/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html"><span style="color:red">[1]</span></a></h3>
        <ul>
          <li><strong>Click</strong> on the upload button in the upper left of the interface.</li>
        </ul>

        <p><img src="images/galaxy_upload.png" alt="" /></p>

        <ul>
          <li>Press <strong>Choose local file</strong> and search for your file.</li>
          <li>Press <strong>Start</strong> and wait for the upload to finish. Galaxy will automatically unpack the file.</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p><strong>Rename</strong> the <em>go.obo</em> file to <strong>GO</strong> and <em>drosophila_gene_association.fb</em> file to <strong>GO annotations Drosophila melanogaster</strong>.</p>

      <blockquote>
        <p>After you upload the files, and if you press the <strong>eye icon</strong> of <em>trapnellPopulation.tab</em> you should look someting like this:
<img src="images/trapnellFile.png" alt="" />
<strong>Figure 2</strong> Trapnell file</p>
      </blockquote>

      <blockquote>
        <p>As you can see, we only have the population file. The file of the study population is missing, so well have to do a data manipulation to obtain it.</p>
      </blockquote>
    </li>
    <li>
      <p>In the tool menu, navigate to <code class="highlighter-rouge">Filter and Sort -&gt; Filter data on any column using simple expressions</code>.</p>

      <blockquote class="comment">
        <h3 id="-comments"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <p>The study sample represents the differentially expressed genes. These were chosen as having an adjusted p-value (last column) smaller than a given threshold. This value is arbitrary, so you may choose the level of significance you want. In this case, we select the genes with an adjusted p-value &lt; 0,05. If you open the <em>trapnellPopulation.tab</em> file, the column 7 corresponds to the <strong>p-values</strong>.</p>
      </blockquote>

      <blockquote>
        <p>Let’s go create the study sample.</p>
      </blockquote>
    </li>
    <li>
      <p><strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: We need to change the following settings:</p>

      <ul>
        <li><strong>Filter</strong>: <code class="highlighter-rouge">trapnellPopulation.tab</code></li>
        <li><strong>With following condition</strong>: <code class="highlighter-rouge">c7 &lt; 0.05</code>
<img src="images/galaxyFilter.png" alt="" /></li>
      </ul>
    </li>
    <li>
      <p>This generate one file called <em>File on data 32</em>. <strong>Rename</strong> to trapnellStudy.</p>

      <blockquote class="comment">
        <h3 id="-comments-1"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <p>Both files have the same information, the little difference between them files is the number of genes. Its important that the genes we have in the study sample must be also in the population sample.</p>
      </blockquote>
    </li>
    <li><strong>GOEnrichment</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <code class="highlighter-rouge">GOEnrichment</code> tool with the four files.
      <ul>
        <li>Use the default options.
<img src="images/galaxyTrapnell.png" alt="" /></li>
      </ul>

      <blockquote class="question">
        <h3 id="-questions"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>After running, what were be the results?</p>
        <details>

<summary>Click to view answers</summary>
This will generate 6 files with the respective default names: MF_Result.txt, BP_Result.txt, CC_Result.txt, MF_Graph, BP_Graph and CC_Graph. Where the three Result files are tables with the statistical tests for each GO Term, and the other three Graph files are graphs displaying the enrichment of the GO terms in the context of their relationships withing the ontology.
</details>
      </blockquote>
    </li>
    <li>
      <p><strong>Rename</strong> files to MF Trapnell, BP Trapnell, CC Trapnell, MF graphTrapnell, BP graphTrapnell and CC graphTrapnell, respectively.
As you can see, the output consists of a table with p values and frequencies. In addition, it also returns, based on the semantics of the GO terms, a graph, where you can view the enrichment results and highlighted enriched ontology branches.</p>

      <blockquote class="comment">
        <h3 id="-comments-2"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <p>For each GO term we obtain a p-value corresponding to a single, independent test. Since we are making multiple similar tests, the probability of at least one of them being a false positive increases. Therefore we need to make multiple tests.</p>
      </blockquote>

      <blockquote class="comment">
        <h3 id="-comments-3"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <p>In general we should correct in cases we’re testing multiple functional. However, the stochastic event (sampling of genes), normally it was already been the subject of statistical testing and multiple test correction. [Ver com o Daniel Faria]</p>
      </blockquote>

      <blockquote class="question">
        <h3 id="-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>How many significant terms do we get?</p>

        <details> 
<summary>Click to view answers</summary>
When we ask how many significant terms, we want to see GO terms that have a p-value &lt; 0.05. According with the results, in Molecular Function we have 5 GO terms, Biological Process we have 43 GO terms and Component Cellular we have 10 GO terms.
</details>
      </blockquote>
      <p>If you press the eye icon of the <em>Molecular Function</em> you should see something like this:</p>
    </li>
  </ol>

  <p><img src="images/mfTrapnell.png" alt="" /></p>

  <blockquote class="comment">
    <h3 id="-comments-4"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
    <p>The ~300 genes should be random. Nonetheless we still have significant terms… [Rever este comentário com o Daniel Faria]</p>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <h3 id="-comments-5"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>

  <p>Let’s go back a little bit, and reopen the trapnellPopulation file. If you go through the file, you’ll see genes with ‘NA’, this means that there are genes in our population that are not differentially expressed.</p>
</blockquote>
<blockquote>

  <p>Let’s go manipulat the <em>trapnellPopulation.tab</em> file to remove the genes that are not differentially expressed. And then see the differences using the GOEnrichment tool.</p>
</blockquote>

<blockquote class="hands_on">
  <h3 id="-hands-on-1"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li><strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: We need to change the following settings:
      <ul>
        <li><strong>Filter</strong>: <code class="highlighter-rouge">trapnellPopulation.tab</code></li>
        <li><strong>With following condition</strong>: <code class="highlighter-rouge">c7 != 'NA'</code>
<img src="images/galaxyFilterNA.png" alt="" /></li>
      </ul>
    </li>
    <li>This generate one file called <em>File on data 32</em>. <strong>Rename</strong> to trapnellNewPopulation.</li>
    <li><strong>GOEnrichment</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <code class="highlighter-rouge">GOEnrichment</code> tool with the new population.
      <ul>
        <li>Use the default options.
<img src="images/galaxyTrapnellNewPop.png" alt="" /></li>
      </ul>
    </li>
    <li><strong>Rename</strong> files to MF New Trapnell, BP New Trapnell, CC New Trapnell, MF New graphTrapnell, BP New graphTrapnell and CC New graphTrapnell, respectively.
Let’s go see again the graph <strong>MF New graphTrapnell</strong>.</li>
  </ol>

  <p><img src="images/mfTrapnellNew.png" alt="" /></p>

  <blockquote class="question">
    <h3 id="-questions-2"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

    <ol>
      <li>How many significant terms do we get?</li>
      <li>Why is different?</li>
    </ol>
    <details>

<summary>Click to view answers</summary>
<ol type="1">
<li> According with the results, in *Molecular Function* and *Biological Process* we have 0 GO terms and *Component Cellular* just 1 GO term. </li>
<li> The non-differentially expressed genes that we exterminate do not have direct connection with the differentially expressed genes of our sample, so we will not get any GO term associated to the ontologies. And also because the genes we select are random genes, and consequently there should be no GO terms associated with the 3 ontologies, which means that the previous result (before the filtering) is not correct.</li>
</ol>
</details>
  </blockquote>
</blockquote>

<h1 id="simplification-of-graphs">Simplification of graphs</h1>

<p>Graphs views are essential, but sometimes the graph view can become overwhelming due to the size of the results. To exemplify this issue, we will next perform functional enrichment analysis using a more realistic dataset from a study using the mouse model organism. The original dataset can be found <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE30352">here</a>. In this <a href="https://www.nature.com/articles/nature10532">study</a>, the authors compared the gene expression of several tissues. Here, we will use results from the comparison between heart and brain.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-2"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <p>For the first exercise we will use as a study sample the differential genes (padjusted&lt;0.05).</p>
  <ol>
    <li><strong>Upload to the Galaxy</strong> the mouse_brain_vs._heart.txt, Mus_musculus_annotations_biomart_e92.tab and mouse_brain_vs_heart.difgenes.txt files.</li>
    <li><strong>Rename</strong> the <em>mouse_brain_vs._heart.txt</em> file to <strong>Mouse population</strong>, <em>Mus_musculus_annotations_biomart_e92.tab</em> file to <strong>GO annotations <em>Mus musculus</em></strong> and <em>mouse_brain_vs_heart.difgenes.txt</em> file to <strong>Mouse diff</strong>.</li>
    <li><strong>GOEnrichment</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <code class="highlighter-rouge">GOEnrichment</code> for the new study sample.
      <ul>
        <li>Select <strong>‘No’</strong> in the Summarize Output option.
<img src="images/galaxyMouseDiff.png" alt="" /></li>
      </ul>
    </li>
    <li>This will generate 6 files, with the respective names: MF_Result.txt, BP_Result.txt, CC_Result.txt, MF_Graph, BP_Graph and CC_Graph. <strong>Rename</strong> to MF tabDiff, BP tabDiff, CC tabDiff, MF grapDiff, BP grapDiff and CC grapDiff, respectively.</li>
    <li>Analyze the table and graph from <em>Biological Process</em>.</li>
  </ol>

  <p><img src="images/bpMouseDiff.png" alt="" /></p>

  <blockquote class="comment">
    <h3 id="-comments-6"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
    <p>As you can see the three graphs are very complex and difficult to analyze.</p>
  </blockquote>
</blockquote>

<p>As you may notice, the number of enriched GO Terms is very high, with graphs that are virtually uninterpretable. One way to avoid this issue is to ignore singletons and skip dependent tests. But even then the graph is still overwhelming.</p>

<p>The Summarize Output option in the GOEnrichment tool simplifies the results further, reducing the complexity while keeping branch information, although at a the cost os some specificity. [Pedir ao Daniel Faria para descrever o método um pouco mais - ver questão de artigo a publicar versus o que se põe aqui?].</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-3"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li><strong>GOEnrichment</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Re-run <code class="highlighter-rouge">GOEnrichment</code> with the same files.
      <ul>
        <li>Use the default options (notice that by default the Summarize Option is on).
<img src="images/galaxyMouseDiffSum.png" alt="" /></li>
      </ul>
    </li>
    <li>Analyze again the table and graph from <em>Biological Process</em>.</li>
  </ol>

  <p><img src="images/bpMouseDiffSum.png" alt="" /></p>

  <blockquote class="question">
    <h3 id="-questions-3"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>
    <ol>
      <li>Are there differences in complexity comparing the graph without the activation of the summarize output option and with?</li>
    </ol>
    <details>

<summary>Click to view answers</summary>
1. Yes, there are differences. As you can see, the activation of the 'summarize' option reduces the size of the graph because, this parameter aims to remove the related terms from the list of enriched GO terms. Resulting in a more summarized graph,  that is, it presents the functions that have a higher frequency in the population. [Com o Daniel Faria, explicar um pouco mais]
</details>
  </blockquote>
</blockquote>

<p>Another alternative tool to reduce the results is to simplify the GO Ontology being used, using GO slims. This gives a much simpler results, but can also gives a substantial loss in specificity, comparing with the Summarize Output. Basically, the GO slims represent cut-down versions of the GO ontologies containing a subset of the terms associated to GO. This gives an ontology without the detail of the specific fine grained terms, and this is the reason for the simpler results.
Let’s use again the results of the mouse, but first we need to use GOSlimmer tool to convert the annotations file.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-4"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li><strong>Upload to the Galaxy</strong> the <a href="http://www.geneontology.org/ontology/subsets/goslim_generic.obo">goslim_generic.obo</a> file.</li>
    <li><strong>Rename</strong> the <em>goslim_generic.obo</em> file to <strong>GO Slim</strong>.</li>
    <li>
      <p><strong>GOSlimmer</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <code class="highlighter-rouge">GOSlimmer</code>.</p>

      <blockquote class="comment">
        <h3 id="-comments-7"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <p>You need to use the GO and GO annotations Mus musculus that you previously upload.</p>
      </blockquote>
    </li>
  </ol>

  <p><img src="images/galaxySlimer.png" alt="" /></p>

  <p>This will generate one file called <strong>Slim Annotations</strong>.</p>
</blockquote>
<blockquote>

  <p>Now we will go use the GOEnrichment tool with the new Slim Annotations file and the same study sample.</p>
</blockquote>

<blockquote class="hands_on">
  <h3 id="-hands-on-5"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li><strong>GOEnrichment</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <code class="highlighter-rouge">GOEnrichment</code>.
      <ul>
        <li>Use the <strong>GOSlim</strong>, <strong>Slim Annotations</strong> and <strong>Mouse population</strong> files.</li>
        <li>Select <strong>‘No’</strong> in the Summarize Output option.
<img src="images/galaxyMouseSlim.png" alt="" /></li>
      </ul>

      <blockquote class="question">
        <h3 id="-questions-4"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <ol>
          <li>What is the difference when you use the GO Slim in data (instead of GO)?</li>
          <li>What is the difference when you apply the GO Slim in data (instead of the Summarize option)?</li>
        </ol>
        <details>

<summary>Click to view answers</summary>
<ol type="1">
<li> In addition to size reduction, the GO Slim shows you an representation of biological information by using high level terms that provide a broad overview of the biology.</li> 
<li> The difference between both is the ontology used. When we apply the summarize option, the GOEnrichment tool will return a summarized output (as we have seen previously), which uses the original/complete&gt;    &gt; &lt;/ol&gt; ontology. When we opted for GO Slim, the original annotation was summarized and because of this it will generate an even more summarized output, and consequently loses a lot of specification unlike the summarize option.</li>
</ol>
</details>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="interpretation-of-the-results">Interpretation of the results</h1>
<p>The interpretation that is performed on the results will depend on the biological information that we intend to extract. Enrichment analysis can be used in validation (e.g., of a protocol for extracting membrane proteins), characterization (e.g., of the effects of a stress in a organism) and elucidation (e.g., of the functions impacted by the knock-out of a transcription factor).
Why is different? There is one important point to keep in mind during the analysis: statistically significant is different from biologically meaningful [Referir outra vez o caso do Trapnell?]. Its possible obtain an some biological or technical insight about the underlying experiment from statistically enriched terms, even if it isn’t readily apparent.</p>

<p>Terms that are very generic tend to be difficult to interpret, since the very specific terms are generally not integrative. It is intended that terms that are sufficiently specific to transmit substantial biological, however, that are generic enough to integrate multiple genes.</p>

<p>And now we go use two study samples, one with overexpressed genes and the other with underexpressed genes, from the same study before.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-6"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li>
      <p><strong>Upload to the Galaxy</strong> the mouseOverexpressed.txt and the mouseUnderexpressed.txt files.</p>

      <blockquote class="question">
        <h3 id="-questions-5"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>How do you know which genes are up- and downregulated? (open the Mouse population file)</p>
        <details>

<summary>Click to view answers</summary>
It is through the logFC values that we derive the information whether the gene is up- or downregulated. If the logFC value is positive it means that the gene is upregulated, and if it is negative the gene is downregulate.
</details>
      </blockquote>

      <blockquote class="comment">
        <h3 id="-comments-8"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>

        <p>To select the up- and downregulated genes, in addition to logFC values, its also necessary define the p-value. In this case was an p-value &lt; 0,01.</p>

      </blockquote>
    </li>
    <li><strong>GOEnrichment</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <code class="highlighter-rouge">GOEnrichment</code> for the both files (<em>mouseOverexpressed.txt</em> and the <em>mouseUnderexpressed.txt</em>).
      <ul>
        <li>Use the <strong>GO</strong>, the <strong>GO annotations <em>Mus musculus</em></strong> and the <strong>Mouse population</strong> files.</li>
        <li>Use the default options.
<img src="images/galaxyMouseOver.png" alt="" />
<img src="images/galaxyMouseUnder.png" alt="" /></li>
      </ul>
    </li>
    <li>
      <p>This will generate 12 files, 6 for each sample, with the respective names: MF_Result.txt, BP_Result.txt, CC_Result.txt, MF_Graph, BP_Graph and CC_Graph. <strong>Rename</strong> according the sample (under- and overexpressed): MF tableUnder, BP tableUnder, CC tableUnder, MF graphUnder, BP graphUnder, CC graphUnder, MF tableOver, BP tableOver, CC tableOver , MF graphOver, BP graphOver and CC graphOver.</p>

      <blockquote class="question">
        <h3 id="-questions-6"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>When analyzing the both tables under and over, what are the related genes, heart or brain?</p>
        <details>

<summary>Click to view answers</summary>
The *BP tableUnder* table regards the genes that are more expressed in the brain than in the heart, therefore the terms are related to brain function. Unlike the *BP tableOver* table, it has the most expressed gjoanapaulas/enes that are related to the heart.
</details>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>
<p>Functional enrichment is a good way to look for certain patterns in populations but their analysis can become a complicated process because of its complexity. And the way to contorne the complexity, is to apply the summarize in ours sets. And thus, we will get an reduction of the complexity and consequently, an major simplicity in the results.</p>



        

        
        <h1>Useful literature</h1>
        <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
        

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i> Help us improve this content!</h3>
            Please take a moment to fill in the Galaxy Training Network
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>.
            Your feedback helps us improve this tutorial and will be considered
            in future revisions.
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> ()!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/functional-enrichment/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
</html>
