<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
    </head>
    <body>
        




<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/sequence-analysis" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Sequence analysis
                        </a>
                    </li>

                    
                        
                        
                        
                    

                    
                        <li class="nav-item dropdown">
                            <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
                                <i class="fa fa-cog" aria-hidden="true"></i> Galaxy Instances
                            </a>
                            <div class="dropdown-menu">
                                
                                    <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/docker" title="Docker image for this tutorial">
                                        <i class="fa fa-ship" aria-hidden="true"></i> Docker image
                                    </a>
                                
                                
                                
                            </div>
                        </li>
                    

                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/sequence-analysis#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i> Help
    </a>
    <div class="dropdown-menu">
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Gitter
        </a>
        <a class="dropdown-item" href="https://biostar.usegalaxy.org/" title="Ask on Biostars">
            Biostars
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/tutorials/genetic-map-rad-seq/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>RAD-Seq to construct genetic maps</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>How to analyze RAD sequencing data for a genetic map study?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>SNP calling from RAD sequencing data</li>
            
            <li>Find and correct haplotypes</li>
            
            <li>Create input files for genetic map building software</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="/training-material/topics//introduction/">Galaxy introduction</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 1/2d</p>
        </blockquote>

        <h1 class="no_toc" id="introduction">Introduction</h1>

<p>This tutorial is based on the analysis originally described in <a href="http://www.genetics.org/content/188/4/799">publication</a>. 
Further information about the pipeline is available from a <a href="http://catchenlab.life.illinois.edu/stacks">dedicated page of the official STACKS website</a>.
Authors describe that they developed a genetic map in the spotted gar and present here data from a single linkage group.
The gar genetic map is an F1 pseudotest cross between two parents and 94 of their F1 progeny. They took the markers that
appeared in one of the linkage groups and worked backwards to provide the raw reads from all of the stacks contributing to that linkage group.</p>

<p>We here proposed to re-analyze these data at least until genotypes determination. Data are already clean so you don’t have to demultiplex it using barcode information through <code class="highlighter-rouge">Process Radtags tool</code>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
    </ol>
  </li>
  <li><a href="#building-loci-using-stacks" id="markdown-toc-building-loci-using-stacks">Building loci using STACKS</a></li>
  <li><a href="#genotypes-determination" id="markdown-toc-genotypes-determination">Genotypes determination</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<h2 id="data-upload">Data upload</h2>

<p>The original data is available at <a href="http://catchenlab.life.illinois.edu/stacks/">STACKS website</a> and the subset used here is findable on <a href="https://zenodo.org/record/1219888#.WtZlK5c6-00">Zenodo</a>.</p>

<blockquote>
  <h3 id="-hands-on-data-upload"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this RAD-seq exercise. If you are not inspired, you can name it “STACKS 1.42 RAD: genetic map” for example…</li>
    <li>
      <p>Import Fasta files from parents and 20 progenies.</p>

      <blockquote>
        <h3 id="-comments"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <p>If you are using the <a href="https://galaxy.genouest.org">GenOuest Galaxy instance</a>, you can load the dataset using ‘Shared Data’ -&gt; ‘Data Libraries’ -&gt; ‘1 Galaxy teaching folder’ -&gt; ‘EnginesOn’ -&gt; ‘RADseq’ -&gt; ‘Genetic map’</p>
      </blockquote>

      <blockquote>
        <h3 id="-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>Paste the following links into the text field
            <div class="highlighter-rouge"><pre class="highlight"><code>https://zenodo.org/record/1219888/files/female
https://zenodo.org/record/1219888/files/male
https://zenodo.org/record/1219888/files/progeny_1
https://zenodo.org/record/1219888/files/progeny_2
https://zenodo.org/record/1219888/files/progeny_3
https://zenodo.org/record/1219888/files/progeny_4
https://zenodo.org/record/1219888/files/progeny_5
https://zenodo.org/record/1219888/files/progeny_6
https://zenodo.org/record/1219888/files/progeny_7
https://zenodo.org/record/1219888/files/progeny_8
https://zenodo.org/record/1219888/files/progeny_9
https://zenodo.org/record/1219888/files/progeny_10
https://zenodo.org/record/1219888/files/progeny_11
https://zenodo.org/record/1219888/files/progeny_12
https://zenodo.org/record/1219888/files/progeny_13
https://zenodo.org/record/1219888/files/progeny_14
https://zenodo.org/record/1219888/files/progeny_15
https://zenodo.org/record/1219888/files/progeny_16
https://zenodo.org/record/1219888/files/progeny_17
https://zenodo.org/record/1219888/files/progeny_18
https://zenodo.org/record/1219888/files/progeny_19
https://zenodo.org/record/1219888/files/progeny_20
</code></pre>
            </div>
          </li>
          <li>Press <strong>Start</strong></li>
        </ul>
      </blockquote>

      <p>As default, Galaxy takes the link as name. It also do not link the dataset to a database or a reference genome.</p>
    </li>
  </ol>

</blockquote>

<h1 id="building-loci-using-stacks">Building loci using STACKS</h1>

<p>Run <code class="highlighter-rouge">Stacks: De novo map</code> Galaxy tool. This program will run <code class="highlighter-rouge">ustacks</code>, <code class="highlighter-rouge">cstacks</code>, and <code class="highlighter-rouge">sstacks</code> on each individual, accounting for the alignments of each read.</p>

<blockquote>
  <h3 id="-comment"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

  <p>Information on <code class="highlighter-rouge">denovo_map.pl</code> and its parameters can be found online: https://creskolab.uoregon.edu/stacks/comp/denovo_map.php.</p>
</blockquote>

<blockquote>
  <p><strong>Stacks: De novo map</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run Stacks selecting the Genetic map usage. Specify each parent as a sample in the appropriate box, then each of the 20 progenies and specify a CP Cross type, 3 for the Minimum number of identical raw reads required to create a stack, 3 for minimum number of identical raw reads required to create a stack in ‘progeny’ individuals, 3 for the number of mismatches allowed between loci when building the catalog and activate the option “remove, or break up, highly repetitive RAD-Tags in the ustacks program”.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovomap_in.png" alt="De novo map input" /></p>
</blockquote>

<blockquote>
  <p>Once Stacks has completed running, you will see 5 new data collections and 8 datasets.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovomap_out.png" alt="The output of de novo map" /></p>

  <p>Investigate the output files: <code class="highlighter-rouge">result.log</code> and <code class="highlighter-rouge">catalog.*</code> (snps, alleles and tags).</p>

  <p>Looking at the first file, <code class="highlighter-rouge">denovo_map.log</code>, you can see the command line used and the start as end execution time.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_top.png" alt="De novo map log file" /></p>

  <p>Then are the different STACKS steps:</p>

  <p><code class="highlighter-rouge">ustacks</code></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_ustacks.png" alt="De novo map:ustacks log" /></p>

  <p><code class="highlighter-rouge">cstacks</code></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_cstacks.png" alt="De novo map: cstacks" /></p>

  <blockquote>
    <h3 id="-question"><i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

    <ol>
      <li>Can you identify the meanning of the number 425?</li>
      <li>
        <p>Looking at the catalog.tags file, identify specific and shared loci from each individual. Count nuber of catalog loci coming from the first individual, from the second and thus find on both parents.</p>

        <details>
<summary>Click to view answers</summary>
<ol type="1">
<li> Here, the catalog is made with 459 tags, 425 coming from the "reference individual", here female. Some of these 425 can be shared with the other parent.</li>
<li>3500</li>
</ol>
</details>
      </li>
    </ol>
  </blockquote>

  <p><code class="highlighter-rouge">sstacks</code></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_sstacks.png" alt="De novo map: sstacks log" /></p>

  <p>Endly, <code class="highlighter-rouge">genotypes</code> is executed. It search for markers identified on the parents and the associate progenies haplotypes. If the first parent have a GA (ex: aatggtgtGgtccctcgtAc) and AC (ex: aatggtgtAgtccctcgtCc) haplotypes, and the second parent only a GA (ex: aatggtgtGgtccctcgtAc) haplotype, STACKS declare a ab/aa marker for this locus. Genotypes program then associate GA to a and AC to b and then scan progeny to determine which haplotype is found on each of them.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_genotypes1.png" alt="De novo map: genotypes log" /></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_genotypes2.png" alt="De novo map: genotypes log" /></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_end.png" alt="De novo map: genotypes log" /></p>

  <p>Finally, 447 loci, markers, are kept to generate the <code class="highlighter-rouge">batch_1.genotypess_1.tsv</code> file. 459 loci are stored on the observed haplotype file <code class="highlighter-rouge">batch_1.haplotypes_1.tsv</code>.</p>

</blockquote>

<h3 id="matches-files">Matches files</h3>
<blockquote>

  <p>Here are <code class="highlighter-rouge">sample1.snps</code> (left) and <code class="highlighter-rouge">sample2.snps</code> (right)</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches1.PNG" alt="De novo map matches files" /></p>

  <p>Catalog_ID (= catalog Stacks_ID) is composed by the <code class="highlighter-rouge">Stack_ID</code> from the “reference” individual, here sample1, but number is different from sample2 <code class="highlighter-rouge">Stack_ID</code>. Thus, in the <code class="highlighter-rouge">catalog.alleles.tsv</code>, the <code class="highlighter-rouge">Stack_ID</code> numbered 3 correspond to the <code class="highlighter-rouge">Stack_ID</code> number 16 from sample2!</p>

  <p>Here you can observe matches files (you maybe have to change the tsv datatype into a tabular one to display correctly the datasets).</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches2.png" alt="Male and female matches files" /></p>

  <p>Considering catalog SNPs 27 &amp; 28, on the 302 catalog locus:</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches_snps.png" alt="De novo map matches considering catalog SNPs" /></p>

  <p>and corresponding catalog haplotypes, 3 on the 4 possible (AA, AT, GT but no GA):</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches_alleles_haplotypes.png" alt="De novo map matches considering catalog haplotypes" /></p>

  <p>heterozygoty is observed on each parent (one ab, the other ac) and we have 19 genotypes on the 22 individuals.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches_markers.png" alt="De novo map macthes: markers" /></p>

  <p>We then can see that Stack_ID 330 for female correspond to the 39 for male:</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches_alleles_male_female.png" alt="De novo map matches: male and female" /></p>

</blockquote>

<h1 id="genotypes-determination">Genotypes determination</h1>
<blockquote>
  <p><strong>Stacks: genotypes</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Re-Run the last step of <code class="highlighter-rouge">Stacks: De novo map</code> pipeline specifying more options as:</p>
  <blockquote>
    <ol>
      <li>
        <p>The genetic map type (ie F1, F2 (left figure, F1xF1), Double Haploid, Back Cross (F1xF0), Cross Pollination (right figure, F1 or F2 but resulting from the croos of pure homozygous parents))</p>

        <p><img src="../../images/RAD2_Genetic_Map/Genetic_map_F2.png" alt="The genetic map type F2" />    <img src="../../images/RAD2_Genetic_Map/Genetic_map_CrossPollination.png" alt="The genetic map CrossPollination" /></p>
      </li>
      <li>
        <p>Genotyping informations export format for input in genetic mapper tools (ie JoinMap, R/qtl, …). To be notice that R/qtl format for an F2 cross type can be an input for MapMaker or Carthagene.</p>
      </li>
      <li>
        <p>Tresholds concernaing a minimal number of progeny and/or minimum stacks depth to consider a locus</p>
      </li>
      <li>
        <p>Automatic corrections. Concerning this last option, it’s possible to to ask genotypes making automatic corrections on some errors as homozygous tags verification in the progeny to be sure a SNP is missing. Thus, if SNP detection model can’t identify a site as heterygous or homozygous, this site is temporarily tagged as homozygous to facilitate the search, by sstacks, of concordance with the loci catalog. If a second allele is detected on the catalog (ie, in parents) is found on a progeny with a weak frequency (&lt;10% of a stack reads number), genotypes program can correct the genotype. Additionally, it will delete a homozygous genotype on a particular individual if locus genotype is supported by less than 5 reads. Thus corrected genotypes are marked uppercase.</p>
      </li>
    </ol>
  </blockquote>

  <p>Here is an example of a locus originally marke as homozygous before automatic correction because on allele is supported by less than 5 reads. After correction, this locus is marked as heterygous.</p>

  <p><img src="../../images/RAD2_Genetic_Map/genotypes_automatic_correction.png" alt="Automatic correction of genotypes" /></p>

  <p>You can re-run <strong>Stacks: genotypes</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: program modifying the number of genotyped progeny to consider a marker and thus be more or less stringent. Compare results.</p>

  <h3 id="genotypestsv-files">Genotypes.tsv files</h3>

  <p>One line by locus, one column by individual (aa, ab, AB if automatic correction applied, bb, bc, …) with observed genotype for each locus:</p>

  <p><img src="../../images/RAD2_Genetic_Map/genotypes_tsv.png" alt="Genotypes.tsv file overview" /></p>

  <h3 id="genotypestxt-files">Genotypes.txt files</h3>

  <p>One line by individual, and for each individual, for each catalog locus, genotype:</p>

  <p><img src="../../images/RAD2_Genetic_Map/genotypes_txt.png" alt="Genotypes.txt file overview" /></p>

  <h3 id="haplotypestsv-files">Haplotypes.tsv files</h3>

  <p>One line by locus, one column by individual (aa, ab, AB if automatic correction applied, bb, bc, …) with observed genotype for each locus:</p>

  <p><img src="../../images/RAD2_Genetic_Map/haplotypes_tsv.png" alt="Haplotypes.tsv file overview" /></p>
</blockquote>

<blockquote>

  <blockquote>
    <h3 id="-question-1"><i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

    <ol>
      <li>The use of the deleverage algorithm allows to not consider loci obtained from merging more than 3 stacks. Why 3 if biologically, you are waiting something related to 2 for diploid organisms?</li>
      <li>
        <p>Re-execute <strong>Stacks: De novo map</strong> pipeline modifying the p-value treshold for the SNP model. What is the difference regarding to unverified haplotypes ?</p>

        <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>This value of 3 is important to use if we don't want to blacklist loci for whom 99.9% of individuals have one and/or the alt allele and 0.01% have a third one, resulting of a sequencing error.</li>
<li>We see a moficiation of the number of unverified haplotypes</li>
</ol>
</details>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we have analyzed real RAD sequencing data to extract useful information, such as genotypes and haplotypes to generate input files for downstream genetic map creation. This approach can be sum up with the following scheme:</p>

<p><img src="../../images/genetic_map_workflow.PNG" alt="The genetic map tutorial workflow" /></p>


        
        <blockquote class="key_points">
            <h3><i class="fa fa-key" aria-hidden="true"></i> Key points</h3>

            <ul>
                
                <li></li>
                
            </ul>
        </blockquote>
        

        
        <h1>Useful literature</h1>
        <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/sequence-analysis#references">here</a>.</p>
        

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i> Help us improve this content!</h3>
            Please take a moment to fill in the Galaxy Training Network
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>.
            Your feedback helps us improve this tutorial and will be considered
            in future revisions.
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Yvan Le Bras)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/tutorials/genetic-map-rad-seq/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
</html>
