<!DOCTYPE html>
<html lang="en">
    <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
    </head>
    <body>
        




<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Transcriptomics
                        </a>
                    </li>

                    
                        
                        
                        
                            <li class="nav-item dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Introduction slides">
                                    <i class="fa fa-slideshare" aria-hidden="true"></i> Introduction slides
                                </a>
                                <div class="dropdown-menu">
                                    
                                        
                                            
                                                <a class="dropdown-item" href="/training-material/topics/transcriptomics/slides/introduction.html">
                                                    Introduction to Transcriptomics
                                                </a>
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                        
                                            
                                        
                                    
                                </div>
                            </li>
                        
                    

                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="https://zenodo.org/record/1185122" title="Links to data">
                            <i class="fa fa-files-o" aria-hidden="true"></i> Input Dataset
                        </a>
                    </li>
                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i> Help
    </a>
    <div class="dropdown-menu">
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Gitter
        </a>
        <a class="dropdown-item" href="https://biostar.usegalaxy.org/" title="Ask on Biostars">
            Biostars
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/ref-based/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>Reference-based RNA-Seq data analysis</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>What are the effects of Pasilla (PS) gene depletion on splicing events?</li>
            
            <li>How to analyze RNA sequencing data using a reference genome?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>Analysis of RNA sequencing data using a reference genome</li>
            
            <li>Analysis of differentially expressed genes</li>
            
            <li>Identification of functional enrichment among differentially expressed genes</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="/training-material/topics//introduction/">Galaxy introduction</a>
                    
                </li>
            
                <li>
                    
                    <a href="/training-material/topics//sequence-analysis/">Quality control</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 1d</p>
        </blockquote>

        <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In the study of <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3032923/">Brooks <em>et al.</em> 2011</a>, the <em>Pasilla</em> (<em>PS</em>) gene, <em>Drosophila</em> homologue of the Human splicing regulators Nova-1 and Nova-2 Proteins, was depleted in <em>Drosophila melanogaster</em> by RNAi. The authors wanted to identify exons that are regulated by <em>Pasilla</em> gene using RNA sequencing data.</p>

<p>Total RNA was isolated and used for preparing either single-end or paired-end RNA-seq libraries for treated (PS depleted) samples and untreated samples. These libraries were sequenced to obtain a collection of RNA sequencing reads for each sample. The effects of <em>Pasilla</em> gene depletion on splicing events can then be analyzed by comparison of RNA sequencing data of the treated (PS depleted) and the untreated samples.</p>

<p>The genome of <em>Drosophila melanogaster</em> is known and assembled. It can be used as reference genome to ease this analysis.  In a reference based RNA-seq data analysis, the reads are aligned (or mapped) against a reference genome, <em>Drosophila melanogaster</em> here, to significantly improve the ability to reconstruct transcripts and then identify differences of expression between several conditions.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li>
<a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
    </ol>
  </li>
  <li>
<a href="#mapping" id="markdown-toc-mapping">Mapping</a>    <ol>
      <li><a href="#mapping-1" id="markdown-toc-mapping-1">Mapping</a></li>
      <li><a href="#inspection-of-the-mapping-results" id="markdown-toc-inspection-of-the-mapping-results">Inspection of the mapping results</a></li>
    </ol>
  </li>
  <li>
<a href="#analysis-of-the-differential-gene-expression" id="markdown-toc-analysis-of-the-differential-gene-expression">Analysis of the differential gene expression</a>    <ol>
      <li><a href="#count-the-number-of-reads-per-annotated-gene" id="markdown-toc-count-the-number-of-reads-per-annotated-gene">Count the number of reads per annotated gene</a></li>
      <li><a href="#identification-of-the-differentially-expressed-features" id="markdown-toc-identification-of-the-differentially-expressed-features">Identification of the differentially expressed features</a></li>
      <li><a href="#visualization-of-the-differentially-expressed-genes" id="markdown-toc-visualization-of-the-differentially-expressed-genes">Visualization of the differentially expressed genes</a></li>
      <li><a href="#analysis-of-the-functional-enrichment-among-the-differentially-expressed-genes" id="markdown-toc-analysis-of-the-functional-enrichment-among-the-differentially-expressed-genes">Analysis of the functional enrichment among the differentially expressed genes</a></li>
    </ol>
  </li>
  <li>
<a href="#inference-of-the-differential-exon-usage" id="markdown-toc-inference-of-the-differential-exon-usage">Inference of the differential exon usage</a>    <ol>
      <li><a href="#count-the-number-of-reads-per-exon" id="markdown-toc-count-the-number-of-reads-per-exon">Count the number of reads per exon</a></li>
      <li><a href="#differential-exon-usage" id="markdown-toc-differential-exon-usage">Differential exon usage</a></li>
    </ol>
  </li>
  <li><a href="#annotation-of-the-result-tables-with-gene-information" id="markdown-toc-annotation-of-the-result-tables-with-gene-information">Annotation of the result tables with gene information</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<h2 id="data-upload">Data upload</h2>

<p>The original data is available at NCBI Gene Expression Omnibus (GEO) under accession number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE18508">GSE18508</a>.</p>

<p>We will look at the 7 first samples:</p>

<ul>
  <li>3 treated samples with <em>Pasilla</em> (PS) gene depletion: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461179">GSM461179</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461180">GSM461180</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461181">GSM461181</a>
</li>
  <li>4 untreated samples: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461176">GSM461176</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461177">GSM461177</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461178">GSM461178</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461182">GSM461182</a>
</li>
</ul>

<p>Each sample constitutes a separate biological replicate of the corresponding condition (treated or untreated). Moreover, two of the treated and two of the untreated samples are from a paired-end sequencing assay, while the remaining samples are from a single-end sequencing experiment.</p>

<p>We have extracted sequences from the Sequence Read Archive (SRA) files to build FASTQ files.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-data-upload">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this RNA-seq exercise</li>
    <li>Import the FASTQ file pairs for
      <ul>
        <li>
<code class="highlighter-rouge">GSM461177</code> (untreated): <code class="highlighter-rouge">GSM461177_1</code> and <code class="highlighter-rouge">GSM461177_2</code>
</li>
        <li>
<code class="highlighter-rouge">GSM461180</code> (treated): <code class="highlighter-rouge">GSM461180_1</code> and <code class="highlighter-rouge">GSM461180_2</code>
</li>
      </ul>

      <p>To import the files, there are two options:</p>
      <ul>
        <li>Option 1: From a shared data library if available (ask your instructor)</li>
        <li>
          <p>Option 2: From <a href="https://dx.doi.org/10.5281/zenodo.1185122">Zenodo</a></p>

          <blockquote class="tip">
            <h3 id="-tip-importing-data-via-links">
<i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Importing data via links</h3>

            <ul>
              <li>Copy the link location</li>
              <li>Open the Galaxy Upload Manager</li>
              <li>Select <strong>Paste/Fetch Data</strong>
</li>
              <li>Paste the link into the text field</li>
              <li>Press <strong>Start</strong>
</li>
            </ul>
          </blockquote>

          <p>You can directly paste:</p>

          <div class="highlighter-rouge">
<pre class="highlight"><code>  https://zenodo.org/record/1185122/files/GSM461177_1.fastqsanger
  https://zenodo.org/record/1185122/files/GSM461177_2.fastqsanger
  https://zenodo.org/record/1185122/files/GSM461180_1.fastqsanger
  https://zenodo.org/record/1185122/files/GSM461180_2.fastqsanger
</code></pre>
          </div>
        </li>
      </ul>
    </li>
    <li>Rename the datasets according to the samples</li>
    <li>
      <p>Check that the datatype is <code class="highlighter-rouge">fastqsanger</code> (<strong>not</strong> <code class="highlighter-rouge">fastq</code>).
If the datatype is <code class="highlighter-rouge">fastq</code>, please change the file type to <code class="highlighter-rouge">fastqsanger</code></p>

      <blockquote class="tip">
        <h3 id="-tip-changing-the-datatype">
<i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the pencil button displayed in your dataset in the history</li>
          <li>Choose <strong>Datatype</strong> on the top</li>
          <li>Select <code class="highlighter-rouge">fastqsanger</code>
</li>
          <li>Press <strong>Save</strong>
</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Add to each database a tag corresponding to the name of the sample (<code class="highlighter-rouge">#GSM461177</code> or <code class="highlighter-rouge">#GSM461180</code>)</p>

      <blockquote class="tip">
        <h3 id="-tip-adding-a-tag">
<i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Adding a tag</h3>
        <ul>
          <li>Click on the dataset</li>
          <li>Click on <strong>Edit dataset tags</strong>
</li>
          <li>
            <p>Add the tag starting with <code class="highlighter-rouge">#</code></p>

            <p>The tags starting with <code class="highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>
          </li>
          <li>Check that the tag is apparing below the dataset name</li>
        </ul>

      </blockquote>
    </li>
  </ol>
</blockquote>

<p>The sequences are raw data from the sequencing machine, without any pretreatments. They need to be assessed for their quality.</p>

<h2 id="quality-control">Quality control</h2>

<p>For quality control, we use similar tools as described in <a href="/training-material/topics/sequence-analysis">NGS-QC tutorial</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> and <a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Trim Galore</a>.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-quality-control">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Quality control</h3>

  <ol>
    <li>
<strong>FastQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run FastQC on the FASTQ files to control the quality of the reads
      <ul>
        <li>“Short read data from your current history”
          <ul>
            <li>Click on “Multiple datasets”</li>
            <li>Select all raw datasets</li>
          </ul>
        </li>
      </ul>

      <blockquote class="tip">
        <h3 id="-tip">
<i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip</h3>

        <p>You can select several files by keeping the CTRL (or COMMAND) key pressed and clicking on the interesting files</p>
      </blockquote>
    </li>
    <li>
      <p>Inspect on the generated webpage for <code class="highlighter-rouge">GSM461177_1</code> sample</p>

      <blockquote class="question">
        <h3 id="-questions">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>What is the read length?</p>

        <details>
   <summary>Click to view answers</summary>
   The read length is 37 bp
   </details>
      </blockquote>
    </li>
    <li>
<strong>MultiQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Aggregate the FastQC reports with
      <ul>
        <li>“Which tool was used generate logs?” to <code class="highlighter-rouge">FastQC</code>
</li>
        <li>“Type of FastQC output?” to <code class="highlighter-rouge">Raw data</code>
</li>
        <li>“FastQC output” to the generated <code class="highlighter-rouge">Raw data</code> files (multiple datasets)</li>
      </ul>
    </li>
    <li>
      <p>Inspect the webpage output from MultiQC</p>

      <blockquote class="question">
        <h3 id="-questions-1">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>What is the quality for the sequences for the different files?</p>

        <details>
   <summary>Click to view answers</summary>
   Everything seems ok for 3 of the files. But for GSM461180_2, the quality seems to decrease quite a lot at the end of the sequences
   </details>
      </blockquote>
    </li>
    <li>
<strong>Trim Galore</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Treat for the quality of sequences by running Trim Galore! with
      <ul>
        <li>“Is this library paired- or single-end?” to <code class="highlighter-rouge">Paired-end</code>
</li>
        <li>First “Reads in FASTQ format” to both <code class="highlighter-rouge">_1</code> fastqsanger datasets (multiple datasets)</li>
        <li>Second “Reads in FASTQ format” to both <code class="highlighter-rouge">_2</code> fastqsanger datasets (multiple datasets)</li>
      </ul>

      <blockquote class="question">
        <h3 id="-questions-2">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>Why do we run Trim Galore! only once on a paired-end dataset and not twice, once for each dataset?</p>

        <details>
<summary>Click to view answers</summary>
Trim Galore can remove sequences if they become too short during the trimming process. For paired-end files Trim Galore! removes entire sequence pairs if one (or both) of the two reads became shorter than the set length cutoff. Reads of a read-pair that are longer than a given threshold but for which the partner read has become too short can optionally be written out to single-end files. This ensures that the information of a read pair is not lost entirely if only one read is of good quality.
</details>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>As the genome of <em>Drosophila melanogaster</em> is known and assembled, we can use this information and map the sequences on this genome to identify the effects of <em>Pasilla</em> gene depletion on splicing events.</p>

<h1 id="mapping">Mapping</h1>

<p>To make sense of the reads, we need to determine to which genes they belong. The first step is to determine their positions within the <em>Drosophila melanogaster</em> genome. This process is known as aligning or ‘mapping’ the reads to a reference.</p>

<blockquote class="comment">
  <h3 id="-comment">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

  <p>Do you want to learn more about the principles behind mapping? Follow our <a href="/training-material/topics/sequence-analysis/">training</a></p>
</blockquote>

<p>Because in the case of a eukaryotic transcriptome, most reads originate from processed mRNAs lacking introns, they cannot be simply mapped back to the genome as we normally do for DNA data. Instead the reads must be separated into two categories:</p>

<ul>
  <li>Reads that map entirely within exons</li>
  <li>Reads that cannot be mapped within an exon across their entire length because they span two or more exons</li>
</ul>

<figure id="figure-1"><img src="../../images/five_type_rna_seq_reads.png" alt="Five types of RNA-seq reads" title="The five types of RNA-seq reads (Figure 1a from Kim et al, Nat Methods, 2015)"><figcaption><span class="figcaption-prefix">Figure 1:</span> The five types of RNA-seq reads (Figure 1a from Kim et al, Nat Methods, 2015)</figcaption></figure>

<p>Spliced mappers have been developed to efficiently map transcript-derived reads against genomes:</p>

<figure id="figure-2"><img src="../../images/splice_aware_alignment.png" alt="Splice-aware alignment" title="Principle of spliced mappers: (1) identification of the reads spanning a single exon, (2) identification of the splicing junctions on the unmapped reads"><figcaption><span class="figcaption-prefix">Figure 2:</span> Principle of spliced mappers: (1) identification of the reads spanning a single exon, (2) identification of the splicing junctions on the unmapped reads</figcaption></figure>

<details>
<summary>Click for more details on the difference spliced mappers</summary>

<p>Several spliced mappers have been developed over the year specially with the explosion of RNA-seq data.</p>

<p><a hef="https://ccb.jhu.edu/software/tophat/index.shtml">TopHat</a> (<a href="https://academic.oup.com/bioinformatics/article/25/9/1105/203994">Trapnell et al, Bioinformatics, 2009</a>) was one of the first tools designed specifically to address this problem. In TopHat reads are mapped against the genome and are separated into two categories: (1) those that map, and (2) those that initially unmapped (IUM). "Piles" of reads representing potential exons are extended in search of potential donor/acceptor splice sites and potential splice junctions are reconstructed. IUMs are then mapped to these junctions.</p>

<figure id="figure-3"><img src="../../images/tophat.png" alt="TopHat" title="TopHat (Trapnell et al, Bioinformatics, 2009)"><figcaption><span class="figcaption-prefix">Figure 3:</span> TopHat (Trapnell et al, Bioinformatics, 2009)</figcaption></figure>

<p>TopHat has been subsequently improved with the development of TopHat2 (<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36">Kim et al, Genome Biology, 2013</a>):</p>

<figure id="figure-4"><img src="../../images/13059_2012_Article_3053_Fig6_HTML.jpg" alt="TopHat2" title="TopHat2 (Kim et al, Genome Biology, 2013)"><figcaption><span class="figcaption-prefix">Figure 4:</span> TopHat2 (Kim et al, Genome Biology, 2013)</figcaption></figure>

<p>To further optimize and speed up spliced read alignment Kim et al (<a href="https://www.nature.com/articles/nmeth.3317">Nat Methods, 2015</a>) developed <a href="https://ccb.jhu.edu/software/hisat2/index.shtml">HISAT</a>. It uses a set of <a href="https://en.wikipedia.org/wiki/FM-index">FM-indices</a> consisting one global genome-wide index and a collection of ~48,000 local overlapping 42 kb indices (~55,000 56 kb indices in HISAT2). This allows to find initial seed locations for potential read alignments in the genome using global index and to rapidly refine these alignments using a corresponding local index:</p>

<figure id="figure-5"><img src="../../images/hisat.png" alt="Hierarchical Graph FM index in HISAT/HISAT2" title="Hierarchical Graph FM index in HiSat/HiSat2 (Kim et al, Nat Methods, 2015)"><figcaption><span class="figcaption-prefix">Figure 5:</span> Hierarchical Graph FM index in HiSat/HiSat2 (Kim et al, Nat Methods, 2015)</figcaption></figure>

<p>A part of the read (blue arrow) is first mapped to the genome using the global FM index. The HISAT then tries to extend the alignment directly utilizing the genome sequence (violet arrow). In (<strong>a</strong>) it succeeds and this read aligned as it completely resides within an exon. In (<strong>b</strong>) the extension hits a mismatch. Now HISAT takes advantage of the local FM index overlapping this location to find the appropriate mapping for the remainder of this read (green arrow). The (<strong>c</strong>) shows a combination these two strategies: the beginning of the read is mapped using global FM index (blue arrow), extended until it reaches the end of the exon (violet arrow), mapped using local FM index (green arrow) and extended again (violet arrow).</p>

<p><a href="https://github.com/alexdobin/STAR">STAR aligner</a> is a fast alternative for mapping RNAseq reads against genome utilizing uncompressed <a href="https://en.wikipedia.org/wiki/Suffix_array">suffix array</a>. It operates in two stages (<a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537">Dobin et al, Bioinformatics, 2013</a>). In the first stage it performs seed search:</p>

<figure id="figure-6"><img src="../../images/star.png" alt="STAR's seed search" title="STAR's seed search (Dobin et al, Bioinformatics, 2013)"><figcaption><span class="figcaption-prefix">Figure 6:</span> STAR's seed search (Dobin et al, Bioinformatics, 2013)</figcaption></figure>

<p>Here a read is split between two consecutive exons. STAR starts to look for a maximum mappable prefix (MMP) from the beginning of the read until it can no longer match continuously. After this point it start to MMP for the unmatched portion of the read (<strong>a</strong>). In the case of mismatches (<strong>b</strong>) and unalignable regions (<strong>c</strong>) MMPs serve as anchors from which to extend alignments</p>

<p>At the second stage STAR stitches MMPs to generate read-level alignments that (contrary to MMPs) can contain mismatches and indels. A scoring scheme is used to evaluate and prioritize stitching combinations and to evaluate reads that map to multiple locations. STAR is extremely fast but requires a substantial amount of RAM to run efficiently.</p>
</details>

<h2 id="mapping-1">Mapping</h2>

<p>We will map our RNA reads to the <em>Drosophila melanogaster</em> genome using STAR.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-spliced-mapping">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Spliced mapping</h3>

  <ol>
    <li>Import the Ensembl gene annotation for <em>Drosophila melanogaster</em> (<code class="highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code>) from the shared data library or from <a href="https://dx.doi.org/10.5281/zenodo.1185122">Zenodo</a> into your current Galaxy history
      <ul>
        <li>Rename the dataset if necessary</li>
        <li>Verify that the datatype is <code class="highlighter-rouge">gtf</code> and not <code class="highlighter-rouge">gff</code>
</li>
      </ul>
    </li>
    <li>
<strong>RNA STAR</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Map your reads on the reference genome with
      <ul>
        <li>“Single-end or paired-end reads” to <code class="highlighter-rouge">Paired-end (as individual datasets)</code>
</li>
        <li>“RNA-Seq FASTQ/FASTA file, forward reads” to the generated <code class="highlighter-rouge">trimmed reads pair 1</code> files (multiple datasets)</li>
        <li>“RNA-Seq FASTQ/FASTA file, reverse reads” to the generated <code class="highlighter-rouge">trimmed reads pair 2</code> files (multiple datasets)</li>
        <li>“Custom or built-in reference genome” to <code class="highlighter-rouge">Use a built-in index</code>
</li>
        <li>“Reference genome with or without an annotation” to <code class="highlighter-rouge">use genome reference without builtin gene-model</code>
</li>
        <li>“Select reference genome” to <code class="highlighter-rouge">Drosophila Melanogaster (dm6)</code>
</li>
        <li>“Gene model (gff3,gtf) file for splice junctions” to the imported <code class="highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code>
</li>
        <li>
          <p>“Length of the genomic sequence around annotated junctions” to <code class="highlighter-rouge">36</code></p>

          <p>This parameter should be length of reads - 1</p>
        </li>
      </ul>
    </li>
    <li>
<strong>MultiQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Aggregate the STAR logs with
      <ul>
        <li>“Which tool was used generate logs?” to <code class="highlighter-rouge">STAR</code>
</li>
        <li>“Type of FastQC output?” to <code class="highlighter-rouge">Log</code>
</li>
        <li>“STAR log output” to the generated <code class="highlighter-rouge">log</code> files (multiple datasets)</li>
      </ul>

      <blockquote class="question">
        <h3 id="-question">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <p>Which percentage of reads were mapped exactly once for both samples?</p>

        <details>
   <summary>Click to view answer</summary>
   More than 83% for GSM461177 and more than 78% for GSM461180
   </details>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p><strong>STAR</strong> generates a BAM file with the mapped reads.</p>

<blockquote class="question">
  <h3 id="-question-1">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

  <ol>
    <li>What is a BAM file?</li>
    <li>
      <p>What does such a file contain?</p>

      <details>
<summary>Click to view answer</summary>
<ol type="1">
<li>a BAM file is the binary version of a SAM file</li>
<li>It contains information about the mapping: for each mapped read, the position on the reference genome, the mapping quality, ...</li>
</ol>
</details>
    </li>
  </ol>
</blockquote>

<h2 id="inspection-of-the-mapping-results">Inspection of the mapping results</h2>

<p>The BAM file contains information about where the reads are mapped on the reference genome. But it is a binary file and with the information for more than 3 million reads encoded in it, it is difficult to inspect and explore the file.</p>

<p>A powerful tool to visualize the content of BAM files is the Integrative Genomics Viewer IGV.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-inspection-of-mapping-results">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Inspection of mapping results</h3>

  <ol>
    <li>
<strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Visualize the aligned reads for <code class="highlighter-rouge">GSM461177</code>
      <ul>
        <li>Click on the STAR BAM output in your history to expand it.</li>
        <li>
          <p>Towards the bottom of the history item, find the line starting with <code class="highlighter-rouge">Display with IGV</code></p>

          <p>This is followed by 2 links:</p>
          <ul>
            <li>option 1: <code class="highlighter-rouge">local</code>. Select this option if you already have IGV installed on your machine</li>
            <li>option 2: <code class="highlighter-rouge">D. melanogaster (dm3)</code>. This will download and launch IGV on your local machine</li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="-comments">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>

        <p>In order for this step to work, you will need to have either IGV or <a href="https://www.java.com/en/download/faq/java_webstart.xml">Java web start</a>
installed on your machine. However, the questions in this section can also be answered by inspecting the IGV screenshots below.</p>

        <p>Check the <a href="https://software.broadinstitute.org/software/igv/AlignmentData">IGV documentation</a> for more information.</p>

      </blockquote>
    </li>
    <li>
      <p><strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Zoom to <code class="highlighter-rouge">chr4:540,000-560,000</code> (Chromosome 4 between 540 kb to 560 kb)</p>

      <blockquote class="question">
        <h3 id="-question-2">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <figure id="figure-7"><img src="../../images/junction_igv_screenshot.png" alt="Screenshot of the IGV view on Chromosome 4" title="Screenshot of IGV on Chromosome 4"><figcaption><span class="figcaption-prefix">Figure 7:</span> Screenshot of IGV on Chromosome 4</figcaption></figure>

        <ol>
          <li>Which information does appear on the top in grey?</li>
          <li>
            <p>What do the connecting lines between some of the aligned reads indicate?</p>

            <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The coverage plot: the sum of mapped reads at each position</li>
<li>They indicate junction events (or splice sites), *i.e.*, reads that are mapped across an intron</li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p><strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Inspect the splice junctions using a <strong>Sashimi plot</strong></p>

      <blockquote class="tip">
        <h3 id="-tip-creation-of-a-sashimi-plot">
<i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Creation of a Sashimi plot</h3>

        <ul>
          <li>Right click on the BAM file</li>
          <li>Select <strong>Sashimi Plot</strong> from the context menu</li>
        </ul>
      </blockquote>

      <blockquote class="question">
        <h3 id="-question-3">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <figure id="figure-8"><img src="../../images/star_igv_sashimi.png" alt="Screenshot of a Sashimi plot of Chromosome 4" title="Screenshot of a Sashimi plot of Chromosome 4"><figcaption><span class="figcaption-prefix">Figure 8:</span> Screenshot of a Sashimi plot of Chromosome 4</figcaption></figure>

        <ol>
          <li>What does the vertical bar graph represent? And the numbered arcs?</li>
          <li>What do the numbers on the arcs mean?</li>
          <li>
            <p>Why do we observe different stacked groups of blue linked boxes at the bottom?</p>

            <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The coverage for each alignment track is plotted as a bar graph. Arcs represent observed splice junctions, *i.e.*, reads spanning introns</li>
<li>The numbers refer to the number of these observed junction reads. </li>
<li>The groups of linked boxes on the bottom represent different transcripts from a single gene differing in the involved exon.</li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>

      <blockquote class="comment">
        <h3 id="-comment-1">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>Check the <a href="https://software.broadinstitute.org/software/igv/Sashimi">IGV documentation on Sashimi plots</a> to find some clues</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>After the mapping, we have the information on where the reads are located on the reference genome. We also know how well they were mapped.</p>

<p>The next step in the RNA-Seq data analysis is quantification of expression level of the genomic features (gene, transcript, exons, …) to be able then to compare several samples for the different expression analysis. The quantification consist into taking each known genomic feature (<em>e.g.</em> gene) of the reference genome and then counting how many reads are mapped on this genomic feature. So, in this step, we start with an information per mapped reads to end with an information per genomic feature.</p>

<blockquote class="comment">
  <h3 id="-comment-2">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

  <p>The quantification depends on the definition of the genomic features of the reference genome, and then on the annotations. We strongly recommend you to use an annotation corresponding to the same version of the reference genome you used for the mapping.</p>
</blockquote>

<p>To identify exons that are regulated by the <em>Pasilla</em> gene, we need to identify genes and exons which are differentially expressed between samples with PS gene depletion and control samples.
In this tutorial, we will then analyze the differential gene expression, but also the differential exon usage.</p>

<p>Did mapping exercise work for you? Great! <img class="emoji" title=":tada:" alt=":tada:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f389.png" height="20" width="20" align="absmiddle"></p>

<h1 id="analysis-of-the-differential-gene-expression">Analysis of the differential gene expression</h1>

<p>We will first investigate the differential gene expression to identify which genes are impacted by the <em>Pasilla</em> gene depletion</p>

<h2 id="count-the-number-of-reads-per-annotated-gene">Count the number of reads per annotated gene</h2>

<p>To compare the expression of single genes between different conditions (<em>e.g.</em> with or without PS depletion), an essential first step is to quantify the number of reads per gene.</p>

<figure id="figure-9"><img src="../../images/gene_counting.png" alt="Counting the number of reads per annotated gene" title="Counting the number of reads per annotated gene"><figcaption><span class="figcaption-prefix">Figure 9:</span> Counting the number of reads per annotated gene</figcaption></figure>

<p>Two main tools could be used for that: <a href="http://htseq.readthedocs.io/en/release_0.9.1/count.html">HTSeq-count</a> (<a href="https://academic.oup.com/bioinformatics/article/31/2/166/2366196">Anders et al, Bioinformatics, 2015</a>) or featureCounts (<a href="https://academic.oup.com/bioinformatics/article/31/2/166/2366196">Liao et al, Bioinformatics, 2014</a>). The second one is considerably faster and requires far less computational resources. We will use it.</p>

<p>In principle, the counting of reads overlapping with genomic features is a fairly simple task. But there are some details that need to be given to featureCounts: for example the strandness…</p>

<h3 id="estimation-of-the-strandness">Estimation of the strandness</h3>

<p>RNAs that are typically targeted in RNAseq experiments are single stranded (<em>e.g.</em>, mRNAs) and thus have polarity (5’ and 3’ ends that are functionally distinct):</p>

<figure id="figure-10"><img src="../../images/dna_rna.png" alt="Relationship between DNA and RNA orientation" title="Relationship between DNA and RNA orientation"><figcaption><span class="figcaption-prefix">Figure 10:</span> Relationship between DNA and RNA orientation</figcaption></figure>

<p>During a typical RNAseq experiment the information about strandedness is lost after both strands of cDNA are synthesized, size selected, and converted into sequencing library. However, this information can be quite useful for the read counting.</p>

<p>Some library preparation protocols create so called <em>stranded</em> RNAseq libraries that preserve the strand information (an excellent overview in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3005310/">Levin et al, Nat Meth, 2010</a>). The implication of stranded RNAseq is that you can distinguish whether the reads are derived from forward- or reverse-encoded transcripts:</p>

<figure id="figure-11"><img src="../../images/stranded_result.png" alt="Stranded RNAseq data look like this" title="How do stranded RNAseq data look like (image from GATC Biotech)"><figcaption><span class="figcaption-prefix">Figure 11:</span> How do stranded RNAseq data look like (image from GATC Biotech)</figcaption></figure>

<p>Depending on the approach and whether one performs single- or paired-end sequencing there are multiple possibilities on how to interpret the results of mapping of these reads onto the genome:</p>

<figure id="figure-12"><img src="../../images/rnaseq_library_type.png" alt="Effects of RNAseq library types" title="Effects of RNAseq library types (adapted from Sailfish documentation)"><figcaption><span class="figcaption-prefix">Figure 12:</span> Effects of RNAseq library types (adapted from Sailfish documentation)</figcaption></figure>

<p>In practice, with Illumina paired-end RNAseq protocols, you are unlikely to uncover many of these possibilities. You will either deal with:</p>

<ul>
  <li>Unstranded RNAseq data</li>
  <li>Stranded RNAseq data produced with Illumina TrueSeq RNAseq kits and <a href="https://nar.oxfordjournals.org/content/37/18/e123">dUTP tagging</a> (<strong>ISR</strong>)</li>
</ul>

<p>This information should usually come with your FASTQ files, ask your sequencing facility! If not, try to find them on the site where you downloaded the data or in the corresponding publication.</p>

<p>Another option is to estimate these parameters with a tool called <strong>Infer Experiment</strong>. This tool takes the output of your mappings (BAM files), takes a subsample of your reads and compares their genome coordinates and strands with those of the reference gene model (from an annotation file). Based on the strand of the genes, it can gauge whether sequencing is strand-specific, and if so, how reads are stranded.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-determining-the-library-strandness">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Determining the library strandness</h3>

  <ol>
    <li>
<strong>Infer Experiment</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Determine the library strandness with:
      <ul>
        <li>“Input .bam file” to the STAR-generated <code class="highlighter-rouge">BAM</code> files (multiple datasets)</li>
        <li>“Reference gene model” to <code class="highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code>
</li>
        <li>“Number of reads sampled from SAM/BAM file (default = 200000)” to <code class="highlighter-rouge">200000</code>
</li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The tool generates one file with:</p>
<ul>
  <li>Paired-end or singled-end library</li>
  <li>Fraction of reads failed to determine</li>
  <li>2 lines
    <ul>
      <li>For single-end
        <ul>
          <li>Fraction of reads explained by “++,–” (<strong>SF</strong> in previous figure)</li>
          <li>Fraction of reads explained by “+-,-+” (<strong>SR</strong> in previous figure)</li>
        </ul>
      </li>
      <li>For paired-end
        <ul>
          <li>Fraction of reads explained by “1++,1–,2+-,2-+” (<strong>SF</strong> in previous figure)</li>
          <li>Fraction of reads explained by “1+-,1-+,2++,2–” (<strong>SR</strong> in previous figure)</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<p>If the fractions in the two last lines are too close to each other, we conclude that this is the library is not specific to a strand specific dataset (<strong>U</strong> in previous figure).</p>

<blockquote class="question">
  <h3 id="-question-4">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

  <ol>
    <li>Which fraction of the reads in the BAM file can be explained assuming which library type for <code class="highlighter-rouge">GSM461177</code>?</li>
    <li>
      <p>Which library type do you choose for both samples?</p>

      <details>
<summary>Click to view answer</summary>
<ol type="1">
<li>Fraction of reads explained by "1++,1--,2+-,2-+": 0.4648 - Fraction of reads explained by "1+-,1-+,2++,2--": 0.4388</li>
<li>The library seems to be of the type unstranded for both samples. </li>
</ol>
</details>
    </li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="-comment-3">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>
  <p>As it is sometimes quite difficult to find out which settings correspond to those of other programs, the following table might be helpful to identify the library type:</p>

  <table>
    <thead>
      <tr>
        <th>Library type</th>
        <th><strong>Infer Experiment</strong></th>
        <th><strong>TopHat</strong></th>
        <th><strong>HISAT2</strong></th>
        <th><strong>htseq-count</strong></th>
        <th><strong>featureCounts</strong></th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Paired-End (PE) - SF</td>
        <td>1++,1–,2+-,2-+</td>
        <td>FR Second Strand</td>
        <td>Second Strand F/FR</td>
        <td>yes</td>
        <td>Forward (1)</td>
      </tr>
      <tr>
        <td>PE - SR</td>
        <td>1+-,1-+,2++,2–</td>
        <td>FR First Strand</td>
        <td>First Strand R/RF</td>
        <td>reverse</td>
        <td>Reverse (2)</td>
      </tr>
      <tr>
        <td>Single-End (SE) - SF</td>
        <td>++,–</td>
        <td>FR Second Strand</td>
        <td>Second Strand F/FR</td>
        <td>yes</td>
        <td>Forward (1)</td>
      </tr>
      <tr>
        <td>SE - SR</td>
        <td>+-,-+</td>
        <td>FR First Strand</td>
        <td>First Strand R/RF</td>
        <td>reverse</td>
        <td>Reverse (2)</td>
      </tr>
      <tr>
        <td>PE, SE - U</td>
        <td>undecided</td>
        <td>FR Unstranded</td>
        <td>default</td>
        <td>no</td>
        <td>Unstranded (0)</td>
      </tr>
    </tbody>
  </table>

</blockquote>

<h3 id="counting">Counting</h3>

<p>We now run <strong>featureCounts</strong> to count the number of reads per annotated gene.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-counting-the-number-of-reads-per-annotated-gene">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Counting the number of reads per annotated gene</h3>

  <ol>
    <li>
<strong>featureCounts</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Count the number of reads per genes using <strong>featureCounts</strong> with
      <ul>
        <li>“Alignment file” to the STAR-generated <code class="highlighter-rouge">BAM</code> files (multiple datasets)</li>
        <li>“Gene annotation file” to <code class="highlighter-rouge">GTF file</code>
</li>
        <li>“Gene annotation file” to <code class="highlighter-rouge">in your history</code>
</li>
        <li>“Gene annotation file” to <code class="highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code>
</li>
        <li>“Output format” to <code class="highlighter-rouge">Gene-ID "\t" read-count (DESeq2 IUC wrapper compatible)</code>
</li>
        <li>Click on “Advanced options”</li>
        <li>“GFF feature type filter” to <code class="highlighter-rouge">exon</code>
</li>
        <li>“GFF gene identifier” to <code class="highlighter-rouge">gene_id</code>
</li>
        <li>“Allow read to contribute to multiple features” to <code class="highlighter-rouge">No</code>
</li>
        <li>“Strand specificity of the protocol” to <code class="highlighter-rouge">Unstranded</code>
</li>
        <li>“Count multi-mapping reads/fragments” to <code class="highlighter-rouge">Disabled; multi-mapping reads are excluded (default)</code>
</li>
        <li>“Minimum mapping quality per read” to <code class="highlighter-rouge">10</code>
</li>
      </ul>
    </li>
    <li>
<strong>MultiQC</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Aggregate the FastQC report with
      <ul>
        <li>“Which tool was used generate logs?” to <code class="highlighter-rouge">featureCounts</code>
</li>
        <li>“Output of FeatureCounts” to the generated <code class="highlighter-rouge">summary</code> files (multiple datasets)</li>
      </ul>

      <blockquote class="question">
        <h3 id="-question-5">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <ol>
          <li>
            <p>How many reads have been assigned to a gene?</p>

            <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>Around 70% of the reads have been assigned to genes: this quantity is correct enough. If it is going below 50%, you should investigate where are mapping your reads (with IGV) and check that the annotation corresponds to the reference genome (version).</li>
</ol>
</details>
          </li>
        </ol>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The main output of <strong>featureCounts</strong> is a big table.</p>

<blockquote class="question">
  <h3 id="-question-6">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

  <ol>
    <li>Which information does the generated table files contain?</li>
    <li>
      <p>Which feature has the most reads mapped on it for both samples?</p>

      <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The useful result file is a tabular file with two columns: the gene id and the number of reads mapped on the corresponding gene</li>
<li>To display the most abundantly detected feature, we need to sort the files with the features and the number of reads mapped to them. This can be done using the Sort tool on the second column and in descending order, which reveals that FBgn0000556 is the feature with the most reads (around 258,000 in GSM461177 and 253,000 in GSM461180) mapped on it.</li>
</ol>
</details>
    </li>
  </ol>
</blockquote>

<h2 id="identification-of-the-differentially-expressed-features">Identification of the differentially expressed features</h2>

<p>In the previous section, we counted reads that mapped to genes for two sample. To be able to identify differential gene expression induced by PS depletion, all datasets (3 treated and 4 untreated) must be analyzed following the same procedure and for the whole genome.</p>

<blockquote class="hands_on">
  <h3 id="-optional-hands-on-re-run-on-the-other-datasets">
<i class="fa fa-pencil" aria-hidden="true"></i> (Optional) Hands-on: Re-run on the other datasets</h3>

  <p>You can do the same process on the other sequence files available on <a href="https://dx.doi.org/10.5281/zenodo.1185122">Zenodo</a></p>

  <ul>
    <li>Paired-end data
      <ul>
        <li>
<code class="highlighter-rouge">GSM461178_1</code> and <code class="highlighter-rouge">GSM461178_2</code>
</li>
        <li>
<code class="highlighter-rouge">GSM461181_1</code> and <code class="highlighter-rouge">GSM461181_2</code>
</li>
      </ul>
    </li>
    <li>Single-end data
      <ul>
        <li><code class="highlighter-rouge">GSM461176</code></li>
        <li><code class="highlighter-rouge">GSM461179</code></li>
        <li><code class="highlighter-rouge">GSM461182</code></li>
      </ul>
    </li>
  </ul>

  <p>This is really interesting to redo on the other datasets, specially to check how the parameters are inferred given the different type of data.</p>
</blockquote>

<p>To save time, we have run the necessary steps for you and obtained 7 count files, available on <a href="https://dx.doi.org/10.5281/zenodo.1185122">Zenodo</a>.</p>

<p>These files contain for each gene of <em>Drosophila</em> the number of reads mapped to it. We could compare the files directly and calculate the extent of differential gene expression, but the number of sequenced reads mapped to a gene depends on:</p>

<ul>
  <li>Its own expression level</li>
  <li>Its length</li>
  <li>The sequencing depth of the sample</li>
  <li>The expression of all other genes within the sample</li>
</ul>

<p>Either for within- or for between-sample comparison, the gene counts need to be normalized. We can then use the Differential Gene Expression (DGE) analysis, whose two basic tasks are:</p>

<ul>
  <li>Estimate the biological variance using the replicates for each condition</li>
  <li>Estimate the significance of expression differences between any two conditions</li>
</ul>

<p>This expression analysis is estimated from read counts and attempts are made to correct for variability in measurements using replicates that are absolutely essential for accurate results. For your own analysis, we advice you to use at least 3, but preferably 5 biological replicates per condition. You can have different number of replicates per condition.</p>

<p><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html"><strong>DESeq2</strong></a> is a great tool for DGE analysis. It takes read counts produced previously, combines them into a big table (with genes in the rows and samples in the columns) and applies size factor normalization:</p>

<ul>
  <li>Computation for each gene of the geometric mean of read counts across all samples</li>
  <li>Division of every gene count by the geometric mean</li>
  <li>Use of the median of these ratios as a sample’s size factor for normalization</li>
</ul>

<p>Multiple factors with several levels can then be incorporated in the analysis. After normalization we can compare, in a statistically reliable way, the response of the expression of any gene to the presence of different levels of a factor.</p>

<p>In our example, we have samples with two varying factors that can explain differences in gene expression:</p>

<ul>
  <li>Treatment (either treated or untreated)</li>
  <li>Sequencing type (paired-end or single-end)</li>
</ul>

<p>Here treatment is the primary factor which we are interested in. The sequencing type is some further information that we know about the data that might affect the analysis. This particular multi-factor analysis allows us to assess the effect of the treatment, while taking the sequencing type into account, too.</p>

<blockquote class="comment">
  <h3 id="-comment-4">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

  <p>We recommend you to add as many factors as you think may affect gene expression in your experiment. It can be the sequencing type like here, but it can also be the manipulation (if different persons are involved in the library preparation), …</p>
</blockquote>

<blockquote class="hands_on">
  <h3 id="-hands-on-determines-differentially-expressed-features">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Determines differentially expressed features</h3>

  <ol>
    <li>Create a new history</li>
    <li>Import the seven count files from <a href="https://dx.doi.org/10.5281/zenodo.1185122">Zenodo</a> or the data library
      <ul>
        <li><code class="highlighter-rouge">GSM461176_untreat_single.counts</code></li>
        <li><code class="highlighter-rouge">GSM461177_untreat_paired.counts</code></li>
        <li><code class="highlighter-rouge">GSM461178_untreat_paired.counts</code></li>
        <li><code class="highlighter-rouge">GSM461179_treat_single.counts</code></li>
        <li><code class="highlighter-rouge">GSM461180_treat_paired.counts</code></li>
        <li><code class="highlighter-rouge">GSM461181_treat_paired.counts</code></li>
        <li><code class="highlighter-rouge">GSM461182_untreat_single.counts</code></li>
      </ul>
    </li>
    <li>
<strong>DESeq2</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>DESeq2</strong> with:
      <ul>
        <li>For “1: Factor”
          <ul>
            <li>“Specify a factor name” to <code class="highlighter-rouge">Treatment</code>
</li>
            <li>“1: Factor level”
              <ul>
                <li>“Specify a factor level” to <code class="highlighter-rouge">treated</code>
</li>
                <li>“Counts file(s)” to the 3 gene count files (multiple datasets) with <code class="highlighter-rouge">treated</code> in name</li>
              </ul>
            </li>
            <li>“2: Factor level”
              <ul>
                <li>“Specify a factor level” to <code class="highlighter-rouge">untreated</code>
</li>
                <li>“Counts file(s)” to the 4 gene count files (multiple datasets) with <code class="highlighter-rouge">untreated</code> in name</li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Click on “Insert Factor” (not on “Insert Factor level”)</li>
        <li>For “2: Factor”
          <ul>
            <li>“Specify a factor name” to <code class="highlighter-rouge">Sequencing</code>
</li>
            <li>“1: Factor level”
              <ul>
                <li>“Specify a factor level” to <code class="highlighter-rouge">PE</code>
</li>
                <li>“Counts file(s)” to the generated count files (multiple datasets) with <code class="highlighter-rouge">paired</code> in name</li>
              </ul>
            </li>
            <li>“2: Factor level”
              <ul>
                <li>“Specify a factor level” to <code class="highlighter-rouge">SE</code>
</li>
                <li>“Counts file(s)” to the generated count files (multiple datasets) with <code class="highlighter-rouge">single</code> in name</li>
              </ul>
            </li>
          </ul>
        </li>
        <li>“Output normalized counts table” to <code class="highlighter-rouge">Yes</code>
</li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><strong>DESeq2</strong> generated 3 outputs</p>

<ul>
  <li>A table with the normalized counts for each genes (rows) and each samples (columns)</li>
  <li>
    <p>A graphical summary of the results, useful to evaluate the quality of the experiment:</p>

    <ol>
      <li>Histogram of <em>p</em>-values for all tests</li>
      <li>
<a href="https://en.wikipedia.org/wiki/MA_plot">MA plot</a>: global view of the relationship between the expression change of conditions (log ratios, M), the average expression strength of the genes (average mean, A), and the ability of the algorithm to detect differential gene expression. The genes that passed the significance threshold (adjusted p-value &lt; 0.1) are colored in red.</li>
      <li>
        <p>Principal Component Analysis (<a href="https://en.wikipedia.org/wiki/Principal_component_analysis">PCA</a>) and the first two axes</p>

        <p>Each replicate is plotted as an individual data point. This type of plot is useful for visualizing the overall effect of experimental covariates and batch effects.</p>

        <blockquote class="question">
          <h3 id="-questions-3">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

          <ol>
            <li>What is the first axis separating?</li>
            <li>
              <p>And the second axis?</p>

              <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>The first axis is seperating the treated samples from the untreated samples, as defined when DESeq2 was launched</li>
<li>The second axis is separating the single-end datasets from the paired-end datasets</li>
</ol>
</details>
            </li>
          </ol>
        </blockquote>
      </li>
      <li>
        <p>Heatmap of sample-to-sample distance matrix: overview over similarities and dissimilarities between samples</p>

        <blockquote class="question">
          <h3 id="-questions-4">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

          <p>How are the samples grouped?</p>

          <details>
   <summary>Click to view answers</summary>
   <ol type="1">    
   <li>They are first grouped depending on the treatment (the first factor) and after on the library type (the second factor), as defined when DESeq2 was launched</li>
   </ol>
   </details>
        </blockquote>
      </li>
      <li>
        <p>Dispersion estimates: gene-wise estimates (black), the fitted values (red), and the final maximum a posteriori estimates used in testing (blue)</p>

        <p>This dispersion plot is typical, with the final estimates shrunk from the gene-wise estimates towards the fitted estimates. Some gene-wise estimates are flagged as outliers and not shrunk towards the fitted value. The amount of shrinkage can be more or less than seen here, depending on the sample size, the number of coefficients, the row mean and the variability of the gene-wise estimates.</p>
      </li>
    </ol>
  </li>
  <li>
    <p>A summary file with the following values for each gene</p>

    <ol>
      <li>Gene identifiers</li>
      <li>Mean normalized counts, averaged over all samples from both conditions</li>
      <li>
        <p>Logarithm (to basis 2) of the fold change</p>

        <p>The log2 fold changes are based on primary factor level 1 vs. factor level 2, hence the order of factor levels is important. For example, for the factor ‘Treatment’, DESeq2 computes fold changes of ‘treated’ samples against ‘untreated’, <em>i.e.</em> the values correspond to up- or downregulation of genes in treated samples.</p>
      </li>
      <li>Standard error estimate for the log2 fold change estimate</li>
      <li>
<a href="https://en.wikipedia.org/wiki/Wald_test">Wald</a> statistic</li>
      <li>
<em>p</em>-value for the statistical significance of this change</li>
      <li>
<em>p</em>-value adjusted for multiple testing with the Benjamini-Hochberg procedure which controls false discovery rate (<a href="https://en.wikipedia.org/wiki/False_discovery_rate">FDR</a>)</li>
    </ol>
  </li>
</ul>

<blockquote class="comment">
  <h3 id="-comment-5">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

  <p>For more information about <strong>DESeq2</strong> and its outputs, you can have a look at <a href="https://www.bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf"><strong>DESeq2</strong> documentation</a>.</p>
</blockquote>

<h2 id="visualization-of-the-differentially-expressed-genes">Visualization of the differentially expressed genes</h2>

<p>We would like now to draw an heatmap of the normalized counts for each sample for the most differentially expressed genes.</p>

<p>We would proceed in several steps</p>
<ul>
  <li>Extract the most differentially expressed genes using the DESeq2 summary file</li>
  <li>Extract the normalized counts of these genes for each sample using the normalized count file generated by DESeq2</li>
  <li>Plot the heatmap of the normalized counts of these genes for each sample</li>
</ul>

<blockquote class="hands_on">
  <h3 id="-hands-on-extract-the-most-differentially-expressed-genes">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Extract the most differentially expressed genes</h3>

  <ol>
    <li>
<strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Extract genes with a significant change in gene expression (adjusted <em>p</em>-value below 0.05) between treated and untreated samples
      <ul>
        <li>“Filter” to the DESeq2 summary file</li>
        <li>“With following condition” to <code class="highlighter-rouge">c7&lt;0.05</code>
</li>
      </ul>

      <blockquote class="question">
        <h3 id="-question-7">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <p>How many genes have a significant change in gene expression between these conditions?</p>

        <details>
<summary>Click to view answers</summary>
We get 1,091 genes (6.21%) with a significant change in gene expression between treated and untreated samples.
</details>
      </blockquote>

      <blockquote class="comment">
        <h3 id="-comment-6">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>The file with the independent filtered results can be used for further downstream analysis as it excludes genes with only few read counts as these genes will not be considered as significantly differentially expressed.</p>
      </blockquote>

      <p>The generated file contains to many genes to get a meaningful heatmap. So we will take only the genes with an absoluted fold change &gt; 2</p>
    </li>
    <li>
<strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Extract genes with an abs(log<sub>2</sub>FC) &gt; 1 (FC stands for “fold change”)
      <ul>
        <li>“Filter” to the differentially expressed genes</li>
        <li>“With following condition” to <code class="highlighter-rouge">abs(c3)&gt;1</code>
</li>
      </ul>

      <blockquote class="question">
        <h3 id="-question-8">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <p>How many genes have been conserved?</p>

        <details>
<summary>Click to view answers</summary>
11.92% (130) of the differentially expressed genes 
</details>
      </blockquote>

      <p>The number of genes is still too high there. So we will take only the 10 most up-regulated and 10 most down-regulated genes</p>
    </li>
    <li>
<strong>Sort</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Sort the genes by log<sub>2</sub>FC
      <ul>
        <li>“Sort Dataset” to the differentially expressed genes with abs(FC) &gt; 2</li>
        <li>“on column” to <code class="highlighter-rouge">3</code>
</li>
        <li>“with flavor” to <code class="highlighter-rouge">Numerical sort</code>
</li>
        <li>“everything in” to <code class="highlighter-rouge">Descending order</code>
</li>
      </ul>
    </li>
    <li>
<strong>Select first lines</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Extract the 10 most up-regulated genes
      <ul>
        <li>“File to select” to the sorted DE genes with abs(FC) &gt; 2</li>
        <li>“Operation” to <code class="highlighter-rouge">Keep first lines</code>
</li>
        <li>“Number of lines” to <code class="highlighter-rouge">10</code>
</li>
      </ul>
    </li>
    <li>
<strong>Select last lines</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Extract the 10 most down-regulated genes
      <ul>
        <li>“Text file” to the sorted DE genes with abs(FC) &gt; 2</li>
        <li>“Operation” to <code class="highlighter-rouge">Keep first lines</code>
</li>
        <li>“Number of lines” to <code class="highlighter-rouge">10</code>
</li>
      </ul>
    </li>
    <li>
<strong>Concatenate datasets</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Concatenated the 10 most up-regulated genes with the 10 most down-regulated genes
      <ul>
        <li>“Datasets to concatenate” to the 10 most up-regulated genes and to the 10 most down-regulated genes</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We now have a table with 20 lines corresponding to the most differentially expressed genes. And for each of the gene, we have its id, its mean normalized counts (averaged over all samples from both conditions), its log<sub>2</sub>FC and other information.</p>

<p>We could plot the log<sub>2</sub>FC for the different genes, but here we would like to look at the heatmap with the read counts for these genes in the different samples. So we need to extract the read counts for these genes.</p>

<p>We will join the normalized count table generated by DESeq with the table we just generated to conserved in the normalized count table only the lines corresponding to the most differentially expressed genes</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-extract-the-normalized-counts-of-most-differentially-expressed-genes-in-the-different-samples">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Extract the normalized counts of most differentially expressed genes in the different samples</h3>

  <ol>
    <li>
<strong>Join two Datasets</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Join the two files
      <ul>
        <li>“Join” to the normalized count table generated by DESeq2</li>
        <li>“using column” to <code class="highlighter-rouge">1</code>
</li>
        <li>“with” to the concatenated file with 10 most up-regulated genes and the 10 most down-regulated genes</li>
        <li>“and column” to <code class="highlighter-rouge">1</code>
</li>
        <li>“Keep lines of first input that do not join with second input” to <code class="highlighter-rouge">No</code>
</li>
        <li>“Keep the header lines” to <code class="highlighter-rouge">Yes</code>
</li>
      </ul>

      <p>The generated files has too many columns: the ones with mean normalized counts, the log<sub>2</sub>FC and other information. We need to remove them</p>
    </li>
    <li>
<strong>Cut</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Conserve the columns with normalized counts
      <ul>
        <li>“Cut columns” to <code class="highlighter-rouge">c1,c2,c3,c4,c5,c6,c7,c8</code>
</li>
        <li>“Delimited by” to <code class="highlighter-rouge">Tab</code>
</li>
        <li>“From” the joined dataset</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We have now a table with 20 lines (the most differentially expressed genes) and the normalized counts for these genes in the 7 samples.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-plot-the-heatmap-of-the-normalized-counts-of-these-genes-for-each-sample">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Plot the heatmap of the normalized counts of these genes for each sample</h3>

  <ol>
    <li>
<strong>heatmap2</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Plot the heatmap with
      <ul>
        <li>“Input should have column headers” to the generated table</li>
        <li>“Advanced - log transformation” to <code class="highlighter-rouge">Log2(value) transform my data</code>
</li>
        <li>“Enable data clustering” to <code class="highlighter-rouge">Yes</code>
</li>
        <li>“Coloring groups” to “Blue to white to red”</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>You should obtain something similar to:</p>

<figure id="figure-13"><img src="../../images/ref_based_most_de_gene_heatmap.png" alt="Heatmap with the normalized counts for the most differentially expressed genes" title="Normalized counts for the most differentially expressed genes"><figcaption><span class="figcaption-prefix">Figure 13:</span> Normalized counts for the most differentially expressed genes</figcaption></figure>

<blockquote class="question">
  <h3 id="-questions-5">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

  <ol>
    <li>Do you observe any tendency in the data?</li>
    <li>What is changing if we select <code class="highlighter-rouge">Plot the data as it is</code> in “Advanced - log transformation”?</li>
    <li>Can you generate an heatmap the normalized counts for the up-regulated genes with FC &gt; 2?</li>
  </ol>

  <details>
<summary>Click to view answers</summary>
<ol type="1">    
  <li>The samples are clustering by treatment. The genes are also clustering based on the counts. 2 genes (FBgn0026562, FBgn00003360)</li>
  <li>The scale is changing and the differences between the genes are not visible anymore</li>
  <li>Extract the genes with logFC &gt; 2 (filter for genes with c3&gt;1 on the summary of the differentially expressed genes) and run heatmap2 on the generated table</li>
</ol>
</details>
</blockquote>

<h2 id="analysis-of-the-functional-enrichment-among-the-differentially-expressed-genes">Analysis of the functional enrichment among the differentially expressed genes</h2>

<p>We have extracted genes that are differentially expressed in treated (with PS gene depletion) samples compared to untreated samples. We would like to know the functional enrichment among the differentially expressed genes.</p>

<p><a href="http://www.geneontology.org/">Gene Ontology (GO)</a> analysis is widely used to reduce complexity and highlight biological processes in genome-wide expression studies, but standard methods give biased results on RNA-seq data due to over-detection of differential expression for long and highly expressed transcripts.</p>

<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/goseq/inst/doc/goseq.pdf">goseq tool</a> provides methods for performing GO analysis of RNA-seq data, taking length bias into account. The methods and software used by goseq are equally applicable to other category based tests of RNA-seq data, such as KEGG pathway analysis.</p>

<p>goseq needs 2 files as inputs:</p>
<ul>
  <li>A tabular file with the differentially expressed genes from all genes assayed in the RNA-seq experiment with 2 columns:
    <ul>
      <li>the Gene IDs (unique within the file)</li>
      <li>True (differentially expressed) or False (not differentially expressed)</li>
    </ul>
  </li>
  <li>A file with information about the length of a gene to correct for potential length bias in differentially expressed genes</li>
</ul>

<blockquote class="hands_on">
  <h3 id="-hands-on-prepare-the-datasets-for-goseq">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Prepare the datasets for GOSeq</h3>

  <ol>
    <li>
<strong>Compute</strong> <i class="fa fa-wrench" aria-hidden="true"></i> with
      <ul>
        <li>“Add expression” to <code class="highlighter-rouge">bool(c7&lt;0.05)</code>
</li>
        <li>“as a new column to” to DESeq summary file</li>
      </ul>
    </li>
    <li>
<strong>Cut</strong> <i class="fa fa-wrench" aria-hidden="true"></i> with
      <ul>
        <li>“Cut columns” to <code class="highlighter-rouge">c1,c8</code>
</li>
        <li>“Delimited by” to <code class="highlighter-rouge">Tab</code>
</li>
        <li>“From” to the previously generated file</li>
      </ul>
    </li>
    <li>
<strong>Change Case</strong> <i class="fa fa-wrench" aria-hidden="true"></i> with
      <ul>
        <li>“From” to the previously generated file</li>
        <li>“Change case of columns” to <code class="highlighter-rouge">c1</code>
</li>
        <li>“Delimited by” to <code class="highlighter-rouge">Tab</code>
</li>
        <li>“To” to <code class="highlighter-rouge">Upper case</code>
</li>
      </ul>

      <p>We just generated the first input for goseq</p>
    </li>
    <li>
<strong>Gene length and GC content</strong> with
      <ul>
        <li>“Select a built-in GTF file or one from your history” to <code class="highlighter-rouge">Use a GTF from history</code>
</li>
        <li>“Select a GTF file” to <code class="highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code>
</li>
        <li>“Select a built-in FASTA or one from your history” to <code class="highlighter-rouge">Use a built-in FASTA</code>
</li>
        <li>“Select a FASTA file” to <code class="highlighter-rouge">Fly (Drosophila melanogaster): dm6 Full</code>
</li>
        <li>“Output length file?” to <code class="highlighter-rouge">Yes</code>
</li>
        <li>“Output GC content file?” to <code class="highlighter-rouge">No</code>
</li>
      </ul>
    </li>
    <li>
<strong>Change Case</strong> <i class="fa fa-wrench" aria-hidden="true"></i> with
      <ul>
        <li>“From” to the previously generated file</li>
        <li>“Change case of columns” to <code class="highlighter-rouge">c1</code>
</li>
        <li>“Delimited by” to <code class="highlighter-rouge">Tab</code>
</li>
        <li>“To” to <code class="highlighter-rouge">Upper case</code>
</li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>We have now the two required files for goseq.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-perform-go-analysis">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Perform GO analysis</h3>

  <ol>
    <li>
<strong>goseq</strong> <i class="fa fa-wrench" aria-hidden="true"></i> with
      <ul>
        <li>“Differentially expressed genes file” to first file generated on previous step</li>
        <li>“Gene lengths file” to second file generated on previous step</li>
        <li>“Gene categories” to <code class="highlighter-rouge">Get categories</code>
</li>
        <li>“Select a genome to use” to <code class="highlighter-rouge">Fruit fly (dm6)</code>
</li>
        <li>“Select Gene ID format” to <code class="highlighter-rouge">Ensembl Gene ID</code>
</li>
        <li>“Select one or more categories” to <code class="highlighter-rouge">GO: Cellular Component</code>, <code class="highlighter-rouge">GO: Biological Process</code>, <code class="highlighter-rouge">GO: Molecular Function</code>
</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>goseq generates a big table with the following columns for each GO term:</p>
<ol>
  <li>
<code class="highlighter-rouge">category</code>: GO category</li>
  <li>
<code class="highlighter-rouge">over_rep_pval</code>: <em>p</em>-value for over representation of the term in the differentially expressed genes</li>
  <li>
<code class="highlighter-rouge">under_rep_pval</code>: <em>p</em>-value for under representation of the term in the differentially expressed genes</li>
  <li>
<code class="highlighter-rouge">numDEInCat</code>: number of differentially expressed genes in this category</li>
  <li>
<code class="highlighter-rouge">numInCat</code>: number of genes in this category</li>
  <li>
<code class="highlighter-rouge">term</code>: detail of the term</li>
  <li>
<code class="highlighter-rouge">ontology</code>: MF (Molecular Function - molecular activities of gene products), CC (Cellular Component - where gene products are active), BP (Biological Process - pathways and larger processes made up of the activities of multiple gene products)</li>
  <li>
<code class="highlighter-rouge">p.adjust.over_represented</code>: <em>p</em>-value for over representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure</li>
  <li>
<code class="highlighter-rouge">p.adjust.under_represented</code>: <em>p</em>-value for over representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure</li>
</ol>

<p>To identify categories significantly enriched/unenriched below some p-value cutoff, it is necessary to use the adjusted <em>p</em>-value.</p>

<blockquote class="question">
  <h3 id="-questions-6">
<i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

  <ol>
    <li>How many GO terms are over represented? Under represented?</li>
    <li>How are the over represented GO terms divided between MF, CC and BP? And for under represented GO terms?</li>
  </ol>

  <details>
<summary>Click to view answers</summary>
<ol type="1">    
  <li>48 GO terms (0.49%) are over represented and 65 (0.66%) - Filter on c8 (over-represented) and c9 (under-represented)</li>
  <li>For over-represented, 31 BP, 6 CC and 10 MF - For under-represented, 36 BP, 25 CC and 3 MF - Group data on column 7 and count on column 1</li>
</ol>
</details>
</blockquote>

<h1 id="inference-of-the-differential-exon-usage">Inference of the differential exon usage</h1>

<p>Next, we would like to know the differential exon usage between treated (PS depleted) and untreated samples using RNA-seq exon counts. We will rework on the mapping results we generated previously.</p>

<p>We will use <a href="https://www.bioconductor.org/packages/release/bioc/html/DEXSeq.html">DEXSeq</a>. DEXSeq detects high sensitivity genes, and in many cases exons, that are subject to differential exon usage. But first, as for the differential gene expression, we need to count the number of reads mapping to the exons.</p>

<h2 id="count-the-number-of-reads-per-exon">Count the number of reads per exon</h2>

<p>This step is similar to the step of <a href="#count-the-number-of-reads-per-annotated-gene">counting the number of reads per annotated gene</a> except that, instead of HTSeq-count, we are using DEXSeq-Count.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-counting-the-number-of-reads-per-exon">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Counting the number of reads per exon</h3>

  <ol>
    <li>
<strong>DEXSeq-Count</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Use the <strong>DEXSeq-Count</strong> to prepare the <em>Drosophila</em> annotations to extract only exons with corresponding gene ids
      <ul>
        <li>“Mode of operation” to <code class="highlighter-rouge">Prepare annotation</code>
</li>
        <li>“GTF file” to <code class="highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code>
</li>
      </ul>

      <p>The output is again a GTF file that is ready to be used for counting</p>
    </li>
    <li>
<strong>DEXSeq-Count</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Count reads using <strong>DEXSeq-Count</strong> with
      <ul>
        <li>“Mode of operation” to <code class="highlighter-rouge">Count reads</code>
</li>
        <li>“Input bam file” to the STAR-generated <code class="highlighter-rouge">BAM</code> files (multiple datasets)</li>
        <li>“DEXSeq compatible GTF file” to the previously generated GTF file</li>
        <li>“Is libray paired end?” to <code class="highlighter-rouge">Yes</code>
</li>
        <li>“Is library strand specific?” to <code class="highlighter-rouge">No</code>
</li>
        <li>“Skip all reads with alignment quality lower than the given minimum value” to <code class="highlighter-rouge">10</code>
</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>DEXSeq generates a file similar to the one generated by featureCounts, but with counts for exons.</p>

<blockquote class="question">
  <h3 id="-question-9">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

  <ol>
    <li>Which exon has the most reads mapped to it for both samples?</li>
    <li>From which gene have these exon been extracted?</li>
    <li>Is there a connection to the previous result obtained with HTSeq-count?</li>
  </ol>

  <details>
<summary>Click to view answers</summary>
FBgn0000556:005 is the exon with the most reads mapped to it for both samples. It is part of FBgn0000556, the feature with the most reads mapped on it (from featureCounts).
</details>
</blockquote>

<h2 id="differential-exon-usage">Differential exon usage</h2>

<p>DEXSeq usage is similar to DESeq2. It uses similar statistics to find differentially used exons.</p>

<p>As for DESeq2, in the previous step, we counted only reads that mapped to exons on chromosome 4 and for only one sample. To be able to identify differential exon usage induced by PS depletion, all datasets (3 treated and 4 untreated) must be analyzed following the same procedure. To save time, we did that for you. The results are available on <a href="https://dx.doi.org/10.5281/zenodo.1185122">Zenodo</a>:</p>

<ul>
  <li>The results of running DEXSeq-count in ‘Prepare annotation’ mode</li>
  <li>Seven count files generated in ‘Count reads’ mode</li>
</ul>

<blockquote class="hands_on">
  <h3 id="-hands-on">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li>Create a new history</li>
    <li>Import the seven exon count files and the annotation GTF file from <a href="https://dx.doi.org/10.5281/zenodo.1185122">Zenodo</a> or the data library
      <ul>
        <li><code class="highlighter-rouge">Drosophila_melanogaster.BDGP6.87.dexseq.gtf</code></li>
        <li><code class="highlighter-rouge">GSM461176_untreat_single.exon.counts</code></li>
        <li><code class="highlighter-rouge">GSM461177_untreat_paired.exon.counts</code></li>
        <li><code class="highlighter-rouge">GSM461178_untreat_paired.exon.counts</code></li>
        <li><code class="highlighter-rouge">GSM461179_treat_single.exon.counts</code></li>
        <li><code class="highlighter-rouge">GSM461180_treat_paired.exon.counts</code></li>
        <li><code class="highlighter-rouge">GSM461181_treat_paired.exon.counts</code></li>
        <li><code class="highlighter-rouge">GSM461182_untreat_single.exon.counts</code></li>
      </ul>
    </li>
    <li>
<strong>DEXSeq</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>DEXSeq</strong> with
      <ul>
        <li>“GTF file created from DEXSeq-Count tool” to <code class="highlighter-rouge">Drosophila_melanogaster.BDGP6.87.dexseq.gtf</code>
</li>
        <li>For “1: Factor”
          <ul>
            <li>“Specify a factor name” to <code class="highlighter-rouge">condition</code>
</li>
            <li>“Specify a factor level” to <code class="highlighter-rouge">treated</code>
</li>
            <li>“Count file for factor level 1” to the exon count files (multiple datasets) with <code class="highlighter-rouge">treated</code> in name</li>
            <li>“Specify a factor level” to <code class="highlighter-rouge">untreated</code>
</li>
            <li>“Count file for factor level 2” to the exon count files (multiple datasets) with <code class="highlighter-rouge">untreated</code> in name</li>
          </ul>
        </li>
        <li>For “2: Factor”
          <ul>
            <li>“Specify a factor name” to <code class="highlighter-rouge">sequencing</code>
</li>
            <li>“Specify a factor level” to <code class="highlighter-rouge">pe</code>
</li>
            <li>“Count file for factor level 1” to the exon count files (multiple datasets) with <code class="highlighter-rouge">paired</code> in name</li>
            <li>“Specify a factor level” to <code class="highlighter-rouge">se</code>
</li>
            <li>“Count file for factor level 2” to the exon count files (multiple datasets) with <code class="highlighter-rouge">single</code> in name</li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="-comment-7">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>Unlike DESeq2, DEXSeq does not allow flexible primary factor names. Always use your primary factor name as “condition”</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>Similarly to DESeq2, DEXSeq generates a table with:</p>

<ol>
  <li>Exon identifiers</li>
  <li>Gene identifiers</li>
  <li>Exon identifiers in the Gene</li>
  <li>Mean normalized counts, averaged over all samples from both conditions</li>
  <li>
    <p>Logarithm (to basis 2) of the fold change</p>

    <p>The log2 fold changes are based on primary factor level 1 vs. factor level 2. The order of factor levels is then important. For example, for the factor ‘Condition’, DESeq2 computes fold changes of ‘treated’ samples against ‘untreated’, <em>i.e.</em> the values correspond to up- or downregulations of genes in treated samples.</p>
  </li>
  <li>Standard error estimate for the log2 fold change estimate</li>
  <li>
<em>p</em>-value for the statistical significance of this change</li>
  <li>
<em>p</em>-value adjusted for multiple testing with the Benjamini-Hochberg procedure which controls false discovery rate (<a href="https://en.wikipedia.org/wiki/False_discovery_rate">FDR</a>)</li>
</ol>

<blockquote class="hands_on">
  <h3 id="-hands-on-1">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li>
      <p><strong>Filter</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Filter</strong> to extract exons with a significant differential usage (adjusted <em>p</em>-value equal or below 0.05) between treated and untreated samples</p>

      <blockquote class="question">
        <h3 id="-question-10">
<i class="fa fa-question-circle" aria-hidden="true"></i> Question</h3>

        <p>How many exons show a significant change in usage between these conditions?</p>

        <details>
<summary>Click to view answers</summary>
We get 38 exons (12.38%) with a significant usage change between treated and untreated samples
</details>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="annotation-of-the-result-tables-with-gene-information">Annotation of the result tables with gene information</h1>

<p>Unfortunately, in the process of counting, we loose all the information of the gene except its identifiant. In order to get the information back to our final counting tables, we can use a tool to make the correspondance between identifiant and annotation.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-2">
<i class="fa fa-pencil" aria-hidden="true"></i> Hands-on:</h3>

  <ol>
    <li>
<strong>Annotate DE(X)Seq result</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Annotate DE(X)Seq result</strong> on a counting table (from DESeq or DEXSeq) using the <code class="highlighter-rouge">Drosophila_melanogaster.BDGP5.78.gtf</code> as annotation file</li>
  </ol>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we have analyzed real RNA sequencing data to extract useful information, such as which genes are up- or downregulated by depletion of the <em>Pasilla</em> gene and which genes are regulated by the <em>Pasilla</em> gene. To answer these questions, we analyzed RNA sequence datasets using a reference-based RNA-seq data analysis approach. This approach can be summarized with the following scheme:</p>

<figure id="figure-14"><img src="../../images/rna_quantification.png" alt="Summary of the analysis pipeline used" title="Summary of the analysis pipeline used"><figcaption><span class="figcaption-prefix">Figure 14:</span> Summary of the analysis pipeline used</figcaption></figure>


        
        <blockquote class="key_points">
            <h3>
<i class="fa fa-key" aria-hidden="true"></i> Key points</h3>

            <ul>
                
                <li>Using a spliced mapping tool for eukaryotic RNA seq data</li>
                
                <li>Running a differential gene expression with taking care of the factors to study</li>
                
                <li>Running a differential exon usage with taking care of the factors to study</li>
                
            </ul>
        </blockquote>
        

        
        <h1>Useful literature</h1>
        <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
        

        <h3>
<i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3>
<i class="fa fa-comments-o" aria-hidden="true"></i> Help us improve this content!</h3>
            Please take a moment to fill in the Galaxy Training Network
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>.
            Your feedback helps us improve this tutorial and will be considered
            in future revisions.
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Bérénice Batut, Mallory Freeberg, Mo Heydarian, Anika Erxleben, Pavankumar Videm, Clemens Blank)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/ref-based/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
</html>
