<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
    </head>
    <body>
        




<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Proteomics
                        </a>
                    </li>

                    
                        
                        
                        
                    

                    
                        <li class="nav-item dropdown">
                            <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
                                <i class="fa fa-cog" aria-hidden="true"></i> Galaxy Instances
                            </a>
                            <div class="dropdown-menu">
                                
                                    <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                                        <i class="fa fa-ship" aria-hidden="true"></i> Docker image
                                    </a>
                                
                                
                                
                            </div>
                        </li>
                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="https://zenodo.org/record/546301" title="Links to data">
                            <i class="fa fa-files-o" aria-hidden="true"></i> Input Dataset
                        </a>
                    </li>
                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i> Help
    </a>
    <div class="dropdown-menu">
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Gitter
        </a>
        <a class="dropdown-item" href="https://biostar.usegalaxy.org/" title="Ask on Biostars">
            Biostars
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>Peptide and Protein ID using SearchGUI and PeptideShaker</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>How to convert LC-MS/MS raw files?</li>
            
            <li>How to identify peptides?</li>
            
            <li>How to identify proteins?</li>
            
            <li>How to evaluate the results?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>Protein identification from LC-MS/MS raw files.</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="/training-material/topics//introduction/">Galaxy introduction</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 45min</p>
        </blockquote>

        <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Identifying the proteins contained in a sample is an important step in any proteomic experiment. However, in most settings, proteins are digested to peptides before the LC-MS/MS analysis. In this so-called “bottom-up” procedure, only peptide masses are measured. Therefore, protein identification cannot be performed directly from raw data, but is a multi-step process:</p>

<ol>
  <li>Raw data preparations</li>
  <li>Peptide-to-Spectrum matching</li>
  <li>Peptide inference</li>
  <li>Protein inference</li>
</ol>

<p>A plethora of different software solutions exists for each step. In this tutorial, we will show how to
use the <a href="http://proteowizard.sourceforge.net/">ProteoWizard</a> tool MSconvert and the <a href="https://openms.de">OpenMS</a> tool <a href="http://ftp.mi.fu-berlin.de/pub/OpenMS/release-documentation/html/TOPP_PeakPickerHiRes.html">PeakPickerHiRes</a> for step 1, and the <a href="https://compomics.com/">Compomics</a> tools <a href="https://compomics.github.io/projects/searchgui.html">SearchGUI</a> and <a href="https://compomics.github.io/projects/peptide-shaker.html">PeptideShaker</a>, for the steps 2-4.</p>

<p>For an alternative ID pipeline using only tools provided by the <a href="https://openms.de">OpenMS software suite</a>, please consult <a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">this tutorial</a>.</p>

<h1 class="no_toc" id="input-data">Input data</h1>

<p>As an example dataset, we will use an LC-MS/MS analysis of HeLa cell lysate published
in <a href="https://www.ncbi.nlm.nih.gov/pubmed/24678044">Vaudel et al., 2014, Proteomics</a>. Detailed information
about the dataset can be found on <a href="https://www.ebi.ac.uk/pride/archive/projects/PXD000674">PRIDE</a>.
For step 2 we will use a validated human Uniprot FASTA database without appended decoys.
If you already completed the tutorial on <a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">Database Handling</a>
you can use the constructed database before the <strong>DecoyDatabase</strong> <i class="fa fa-wrench" aria-hidden="true"></i> step. You can find a prepared database, as well as the input proteomics data in different file formats on <a href="https://zenodo.org/record/796184">Zenodo</a>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#preparing-raw-data" id="markdown-toc-preparing-raw-data">Preparing Raw Data</a></li>
  <li><a href="#peptide-and-protein-identification" id="markdown-toc-peptide-and-protein-identification">Peptide and Protein Identification</a></li>
  <li><a href="#analysis-of-contaminants" id="markdown-toc-analysis-of-contaminants">Analysis of Contaminants</a></li>
  <li><a href="#evaluation-of-peptide-and-protein-ids" id="markdown-toc-evaluation-of-peptide-and-protein-ids">Evaluation of Peptide and Protein IDs</a></li>
</ol>

</blockquote>

<h1 id="preparing-raw-data">Preparing Raw Data</h1>

<p>Raw data conversion is the first step of any proteomic data analysis. The most common converter is MSConvert from the <a href="http://proteowizard.sourceforge.net/">ProteoWizard software suite</a>, the format to convert to is mzML. Search GUI needs <code class="highlighter-rouge">mgf</code> format as input, but as we need the <code class="highlighter-rouge">mzML</code> format for several other tasks, we will convert to <code class="highlighter-rouge">mzML</code> first. Due to licensing reasons, MSConvert runs only on windows systems and will not work on most Galaxy servers.</p>

<p>Depending on your machine settings, raw data will be generated either in profile mode or centroid mode. For most peptide search engines, the MS2 data have to be converted to centroid mode, a process called “peak picking” or “centroiding”. 
Machine vendors offer algorithms to extract peaks from profile raw data. This is implemented in <strong><em>msconvert</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> and can be run in parallel to the mzML conversion. However, the OpenMS tool <strong><em>PeakPickerHiRes</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> is reported to generate slightly better results (<a href="https://www.ncbi.nlm.nih.gov/pubmed/17094243">Lange et al., 2006, Pac Symp Biocomput</a>) and is therefore recommended for quantitative studies (<a href="https://www.ncbi.nlm.nih.gov/pubmed/19953549">Vaudel et al., 2010, Proteomics</a>).
If your data were generated on a low resolution mass spectrometer, use <strong><em>PeakPickerWavelet</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> instead.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-file-conversion-and-peak-picking"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-On: File Conversion and Peak Picking</h3>

  <p>We provide the <a href="https://zenodo.org/record/796184">input data</a> in the original <code class="highlighter-rouge">raw</code> format and also already converted to <code class="highlighter-rouge">mgf</code> and <code class="highlighter-rouge">mzML</code> file formats. If <strong><em>msconvert</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> does not run on your Galaxy instance, please download the preconverted <code class="highlighter-rouge">mzML</code> as an input.</p>

  <ol>
    <li>Create a new history for this Peptide and Protein ID exercise.</li>
    <li>Load the example dataset into your history from Zenodo: <a href="https://zenodo.org/record/892005/files/qExactive01819.raw">raw</a> <a href="https://zenodo.org/record/892005/files/qExactive01819_profile.mzml">mzML</a></li>
    <li>Rename the dataset to something meaningful.</li>
    <li>(<em>optional</em>) Run <strong><em>msconvert</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> on the test data to convert to the <code class="highlighter-rouge">mzML</code> format.</li>
    <li>Run <strong><em>PeakPickerHiRes</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> on the resulting file. Click <code class="highlighter-rouge">+ Insert param.algorithm_ms_levels</code> and change the entry to “2”. Thus, peak picking will only be performed on MS2 level.</li>
    <li>Run <strong><em>FileConverter</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> on the picked mzML. In the <strong>Advanced Options</strong> set the <strong>Output file type</strong> to <code class="highlighter-rouge">mgf</code>.</li>
  </ol>

  <blockquote class="comment">
    <h3 id="-comment-local-use-of-msconvert"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment: Local Use of MSConvert</h3>
    <p>The vendor libraries used by MSConvert are only licensed for Windows systems and are therefore rarely implemented in Galaxy instances. If <strong><em>msconvert</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> is not available in your Galaxy instance, please install the software on a Windows computer and run the conversion locally. You can find a detailed description of the necessary steps <a href="https://compomics.com/bioinformatics-for-proteomics/identification/">here</a> (“Peak List Generation”). Afterwards, upload the resulting mzML file to your Galaxy history.</p>
  </blockquote>
</blockquote>

<h1 id="peptide-and-protein-identification">Peptide and Protein Identification</h1>
<p>MS/MS experiments identify peptides by isolating them and subsequently colliding them with a gas for fragmentation. This method generates a spectrum of peptide fragment masses for each isolated peptide - an MS2 spectrum. To find out the peptide sequences, the MS2 spectrum is compared to a theoretical spectrum generated from a protein database. This step is called peptide-to-spectrum (also: spectrum-to-sequence) matching. Accordingly, a peptide that is successfully matched to a sequence is termed PSM (Peptide-Spectrum-Match). There can be multiple PSMs per peptide, if the peptide was fragmented several times. Different peptide search engines have been developed to fulfill the matching procedure.</p>

<p>It is generally recommended to use more than one peptide search engine and use the combined results for the final peptide inference (<a href="https://www.ncbi.nlm.nih.gov/pubmed/23720762">Shteynberg et al., 2013, Mol. Cell. Proteomics</a>). Again, there are several software solutions for this, e.g. iProphet (TPP) or ConsensusID (OpenMS). In this tutorial we will use <strong><em>Search GUI</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i>, as it can automatically search the data using several search engines. Its partner tool <strong><em>Peptide Shaker</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> is then used to combine and evaluate the search engine results.</p>

<p>In bottom-up proteomics, it is necessary to combine the identified peptides to proteins. This is not a trivial task, as proteins are redundant to some degree. Thus, not every peptide can be assigned to only one protein. Luckily, the <strong><em>Peptide Shaker</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> already takes care of protein inference and even gives us some information on validity of the protein IDs. We will discuss validation in a <a href="#evaluation-of-peptide-and-protein-ids">later step</a> of this tutorial.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-peptide-and-protein-identification"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-On: Peptide and Protein Identification</h3>

  <ol>
    <li>Copy the prepared protein database from the tutorial <a href="../database-handling/tutorial.html">Database Handling</a> into your current history by using the multiple history view or upload the ready-made database from this <a href="https://zenodo.org/record/892005/files/Human_database_%28cRAP_and_Mycoplasma_added%29.fasta">link</a>.</li>
    <li>Open <strong><em>Search GUI</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> to search the mgf file against the protein database. In the <strong><code class="highlighter-rouge">Search Engine Options</code></strong> select <code class="highlighter-rouge">X!Tandem</code> and <code class="highlighter-rouge">MS-GF+</code>. In the <strong><code class="highlighter-rouge">Protein Modification Options</code></strong> add the <strong><code class="highlighter-rouge">Fixed Modifications</code></strong>: <code class="highlighter-rouge">Carbamidomethylation of C</code> and the <strong><code class="highlighter-rouge">Variable Modifications</code></strong>: <code class="highlighter-rouge">Oxidation of M</code>.</li>
    <li>Run <strong><em>Peptide Shaker</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> on the Search GUI output. Enable the following outputs: <code class="highlighter-rouge">Zip File for import to Desktop App</code>, <code class="highlighter-rouge">mzidentML File</code>, <code class="highlighter-rouge">PSM Report</code>, <code class="highlighter-rouge">Peptide Report</code>, <code class="highlighter-rouge">Protein Report</code>.</li>
  </ol>

  <blockquote class="comment">
    <h3 id="-comment-search-gui-parameters"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment: Search GUI Parameters</h3>
    <p>We ran <strong><em>Search GUI</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> with default settings. When you are processing files of a different experiment, you may need to adjust some of the parameters.
<strong>Search GUI</strong> bundles numerous peptide search engines for matching MS/MS to peptide sequences within a database. In practice, using 2-3 different search engines offers high confidence while keeping analysis time reasonable.
The <strong><code class="highlighter-rouge">Precursor Options</code></strong> have to be adjusted to the mass spectrometer which was used to generate the files. The default settings fit a  high resolution Orbitrap instrument.
In the <strong><code class="highlighter-rouge">Advanced Options</code></strong> you may set much more detailed settings for each of the used search engines. When using X!Tandem, we recommend to switch off the advanced X!Tandem options <strong><code class="highlighter-rouge">Noise suppression</code></strong>, <strong><code class="highlighter-rouge">Quick Pyrolidone</code></strong> and <strong><code class="highlighter-rouge">Quick Acetyl</code></strong>. When using MSGF, we recommend to select the correct <strong><code class="highlighter-rouge">Instrument type</code></strong>.</p>
  </blockquote>

  <blockquote class="comment">
    <h3 id="-comment-peptideshaker-outputs"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment: PeptideShaker Outputs</h3>
    <p>Peptide Shaker offers a variety of outputs. 
The <code class="highlighter-rouge">Zip File for import to Desktop App</code> can be downloaded to view and evaluate the search results in the Peptide Shaker viewer (<a href="https://compomics.github.io/projects/peptide-shaker.html">Download</a>). 
The several <code class="highlighter-rouge">Reports</code> contain tabular, human-readable information. 
Also, an <code class="highlighter-rouge">mzidentML</code> (= <code class="highlighter-rouge">mzid</code>) file can be created that contains all peptide sequence matching information and can be utilized by compatible downstream software.
The <code class="highlighter-rouge">Certificate of Analysis</code> provides details on all parameters settings of both Search GUI and Peptide Shaker used for the analysis.</p>
  </blockquote>

  <blockquote class="question">
    <h3 id="-questions"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions:</h3>
    <ol>
      <li>How many peptides were identified? How many proteins?</li>
      <li>How many peptides with oxidized methionine were identified?</li>
    </ol>

    <details>
 <summary>Click to view answers</summary>
  <ol type="1">
    <li> You should have identified 3,325 peptides and 1,170 proteins.</li>
    <li> 328 peptides contain an oxidized methionine (MeO). To get to this number, you can use ***Select*** <i class="fa fa-wrench" aria-hidden="true"></i> on the Peptide Report and search for either "Oxidation of M" or "M\&lt;ox\&gt;".</li>
  </ol>
 </details>
  </blockquote>
</blockquote>

<h1 id="analysis-of-contaminants">Analysis of Contaminants</h1>
<p>The FASTA database used for the peptide to spectrum matching contained some entries that were not expected to stem from the HeLa cell lysate, but are common contaminations in LC-MS/MS samples. The main reason to add those is to avoid false assignment of the spectra to other proteins. However, it also enables you to check for contaminations in your samples. <strong>CAVE:</strong> in human samples, many proteins that are common contaminants may also stem from the real sample. The real source can never be determined for sure.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-analysis-of-contaminants"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-On: Analysis of Contaminants</h3>

  <ol>
    <li>Run <strong><em>Select</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> on the Peptide Shaker Protein Report to select all lines that match the pattern “CONTAMINANT”.</li>
    <li>Remove all contaminants from your protein list by running <strong><em>Select</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> on the Peptide Shaker Protein Report. Select only those lines that <strong>do not</strong> match the pattern “CONTAMINANT”.</li>
  </ol>

  <blockquote class="question">
    <h3 id="-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>
    <ol>
      <li>Which contaminants did you identify? Where do these contaminations come from?</li>
      <li>What other sources of contaminants exist?</li>
      <li>How many mycoplasma proteins did you identify? Does this mean that the analyzed HeLa cells were infected with mycoplasma?</li>
      <li>How many false positives do we expect in our list? How many of these are expected to match mycoplasma proteins?</li>
    </ol>

    <details>
 <summary>Click to view answers</summary>
  <ol type="1">
    <li> TRY_BOVIN is bovine trypsin. It was used to degrade the proteins to peptides. ALBU_BOVIN is bovine serum albumin. It is added to cell culture medium in high amounts.</li>
    <li> Contaminants often stem from the experimenter, these are typically keratins or other high-abundant human proteins. Basically any protein present in the room of the mass spectrometer might get into the ion source, if it is airborne. As an example, sheep keratins are sometimes found in proteomic samples, stemming from clothing made of sheep wool.</li>
    <li> There should be five mycoplasma proteins in your protein list. However, all of them stem from different mycoplasma species. Also, every protein was identified by one peptide only. You can see this in column 17-19 of your output. These observations make it very likely that we are facing false positives here.</li>
    <li> As we were allowing for a false discovery rate of 1 %, we would expect 12 false positive proteins in our list.
    False positives are expected to be randomly assigned to peptides in the FASTA database. Our database consists of about 20,000 human proteins and 4,000 mycoplasma proteins. Therefore, we would expect 17 % (= 2) of all false positives matching to mycoplasma proteins.</li>
  </ol>
 </details>
  </blockquote>
</blockquote>

<h1 id="evaluation-of-peptide-and-protein-ids">Evaluation of Peptide and Protein IDs</h1>
<p><strong><em>Peptide Shaker</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> provides you with validation results for the identified PSM, peptides and proteins. It classifies all these IDs in the categories “Confident” or “Doubtful”. On each level, the meaning of these terms differs to some extent:</p>

<ul>
  <li><strong>PSMs</strong> are marked as “Doubtful” when the measured MS2 spectrum did not fit well to the theoretical spectrum.</li>
  <li><strong>Peptides</strong> have a combined scoring of their PSMs. They are marked as “Doubtful”, when the score is below a set threshold. The threshold is defined by the false discovery rate (FDR).</li>
  <li><strong>Proteins</strong> are marked as “Doubtful”, when they were identified by only a single peptide or when they were identified solely by “Doubtful” peptides.</li>
</ul>

<blockquote class="hands_on">
  <h3 id="-hands-on-evaluation-of-peptide-and-protein-ids"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-On: Evaluation of Peptide and Protein IDs</h3>

  <ol>
    <li>Remove all “Doubtful” proteins from your protein list by running <strong><em>Select</em></strong> <i class="fa fa-wrench" aria-hidden="true"></i> on the Peptide Shaker Protein Report. Select only those lines that <strong>do not</strong> match the pattern “Doubtful”.</li>
  </ol>

  <blockquote class="question">
    <h3 id="-questions-2"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions:</h3>
    <ol>
      <li>How to exclude mycoplasma proteins?</li>
      <li>How many “Confident” non-contaminant proteins were identified?</li>
    </ol>

    <details>
 <summary>Click to view answers</summary>
  <ol type="1">
      <li> Add another ***Select*** <i class="fa fa-wrench" aria-hidden="true"></i> matching the pattern "HUMAN".</li>
  <li> You should have identified 582 human non-contaminant proteins that were validated to be "Confident".&lt;\li&gt;
  &lt;/ol&gt;
 &lt;/details&gt;
</li></ol></details>
  </blockquote>
</blockquote>

<h1 class="no_toc" id="premade-workflow">Premade Workflow</h1>

<p>A premade workflow for this tutorial can be found <a href="workflows/wf_proteinID_SG_PS.ga">here</a></p>

<h1 class="no_toc" id="further-reading">Further Reading</h1>

<ul>
  <li><a href="https://compomics.com/bioinformatics-for-proteomics/">Search GUI and Peptide Shaker tutorials at Compomics</a></li>
  <li><a href="https://drive.google.com/file/d/0B6bIeOvjBkbWVnhMLWxXdGVUY3M/view">Using Search GUI and Peptide Shaker in Galaxy</a></li>
</ul>


        
        <blockquote class="key_points">
            <h3><i class="fa fa-key" aria-hidden="true"></i> Key points</h3>

            <ul>
                
                <li>LC-MS/MS raw files have to be locally converted to mgf/mzML prior to further analysis on most Galaxy servers.</li>
                
                <li>SearchGUI can be used for running several peptide search engines at once.</li>
                
                <li>PeptideShaker can be used to combine and evaluate the results, and to perform protein inference.</li>
                
            </ul>
        </blockquote>
        

        
        <h1>Useful literature</h1>
        <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
        

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i> Help us improve this content!</h3>
            Please take a moment to fill in the Galaxy Training Network
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>.
            Your feedback helps us improve this tutorial and will be considered
            in future revisions.
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Florian Christoph Sigloch, Björn Grüning)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
</html>
