<<<<<<< HEAD
---
layout: introduction_slides
topic_name: metagenomics
tutorial_name: introduction
logo: "GTN"
=======
<!DOCTYPE html>
<html>






  <head>
    <meta charset="utf-8">
    <title>Introduction to metagenomics</title>
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css" id="theme">
  </head>
  <body>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-footer"><span>

<img src="/training-material/assets/images/GTN-60px.png" alt="Galaxy Training Network" style="height: 40px;"/>

</span></div>

---

# Introduction to metagenomics


---

  ## Requirements

  Before diving into this slide deck, we recommend you to have a look at:

  
    
  - [Galaxy introduction](/training-material/topics//introduction/)
    
  

  
    
  


>>>>>>> f1b01b7ef32ec4f29a7f797f014dae1d28ea2906
---

## Metagenomics

---

## Overview

- Applications of metagenomics
- Amplicon (16S) Approach
- Shotgun Approach


---

## Applications

- **Environmental Studies**
- **Clinical**
  - Humans are full of microorganisms affecting health, drug efficacy, etc


.image-75[ ![](../images/human_microbiome.png) ]


---

## Amplicon


- Targetted approach (e.g. 16S/18S rRNA gene)
- Amplifies bacteria, not host or environmental fungi, plants

![](../images/16S_gene.png)

---

## Amplicon

- Highly conserved gene: easy to target across all bacteria
- With variable regions: distinguish between genus

![](../images/16S_variableregions.jpg)

???
V1, V2 etc are the variable regions

---

## Amplicon

- Pros
  - Well-established
  - Inexpensive ($50-$100/sample)
- Cons
 - V-region choice can bias results
 - Is based on a very well conserved gene, making it hard to resolve species and strains


???
18S gene also amplifies archaea

---

## Shotgun metagenomic

<<<<<<< HEAD
- Aims to sequence the "whole" metagenome
=======
- Aims to sequence the &quot;whole&quot; metagenome
>>>>>>> f1b01b7ef32ec4f29a7f797f014dae1d28ea2906

- Pros:
  - Not biased by amplicon primer set
  - Not limited by conservation of the amplicon
  - Can also provide functional information

- Cons:
  - Environmental contamination, including host
  - More expensive ($1000+/sample)
  - Complex data analysis
    - Requires high performance computing, high memory, high compute capacity

---


## End-to-End

![](../images/16S_end-to-end.png)

Every step in this process can have serious impact on the results

---

## Bioinformatics

![](../images/so_much_data.png)

```
Roche 454 GS:    ~ 100.000 reads
Illumina MiSeq:  ~ 25.000.000 reads
Shotgun:             ~ ? reads
```

---

## Analysis pipelines

![](../images/metagenomics_pipeline.png)

---

## Pre-processing

![](../images/preprocessing.png)

- There are a lot of ways to filter and trim your data
- Trade-off between quality and amount of information retained

---

## Chimera Removal

- During PCR multiple sequences can combine to form a hybrid
- Must be removed from your data for better results

![](../images/chimeras.png)

---

## OTU Clustering

Cluster on 97% sequence similarity for genus-level differentiation

.pull-left[ .image-100[ ![](../images/clustering.png) ]]

.pull-right[.image-100[ ![](../images/otu.png) ]]

---

## Search marker database and taxonomy assignment

- Homology with reference databases
  - Amplicon

    ![](../images/silva-greengenes.png)

  - Shotgun: MetaPhlAn2 database
      - ~1M unique clade-specific marker genes
      - ~17,000 reference genomes (bacterial and archaeal, viral and eukaryotic)

- Accuracy depends on quality and completeness of database used
- Databases are inevitable incomplete

---

## Search functional database

- Databases to identify gene families
    - UniRef50
    - UniRef90
- Grouping in other functional categories
    - MetaCyc Reactions
    - KEGG Orthogroups (KOs)
    - Pfam domains
    - Level-4 enzyme commission (EC) categories
    - EggNOG (including COGs)
    - Gene Ontology (GO)
    - Informative GO
- Pathway reconstruction

---

## Results: OTU table

![](../images/otutable.png)


---

## Results: Visualizations

- **Krona**
  - interactive exploration of sample taxonomy

![](../images/krona.png)

---

## Results: Visualizations

- **Phinch**
  - BIOM file input
  - various visualizations
  - multi-sample data

![](../../../shared/images/phinch_overviewpage.png)
<<<<<<< HEAD
=======


---



## Related tutorials


    

    
        
    

    
        
    



---

## Thank you!

This material is the result of a collaborative work. Thanks the [Galaxy Training Network](https://wiki.galaxyproject.org/Teach/GTN) and all the contributors  (Bérénice Batut, Saskia Hiltemann) !


<img src="/training-material/assets/images/GTN.png" alt="Galaxy Training Network" style="height: 100px;"/>



.footnote[Found a typo? Something is wrong in this tutorial? <br/>Edit it on [GitHub](https://github.com/galaxyproject/training-material/tree/master/topics/metagenomics/slides/introduction.html)]

    </textarea>
    <script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript">
    </script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
    </script>
  </body>
</html>
>>>>>>> f1b01b7ef32ec4f29a7f797f014dae1d28ea2906
